<!DOCTYPE html><html lang="en-us" >

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 4.8.0">

  

  
  
  
  
  
    
    
    
  
  

  

  
  
  
    
  
  <meta name="description" content="
简介
Sabeti在2002年的时候提出了一种叫做Extended Haplotype Homozygosity (EHH)的统计方法，用来扫描全基因组范围内的选择印记。在此基础上又衍生出了iHS、XP-EHH以及Rsb等统计方法。R包rehh实现了这几种统计方法的封装。">

  
  <link rel="alternate" hreflang="en-us" href="https://taoyan.netlify.app/post/2019-08-12.rehh%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%80%89%E6%8B%A9%E5%8D%B0%E8%AE%B0%E6%89%AB%E6%8F%8F/">

  


  
  
  
  <meta name="theme-color" content="#06805E">
  

  
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.0-1/css/all.min.css" integrity="sha256-4w9DunooKSr3MFXHXWyFER38WmPdm361bQS/2KUWZbU=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css" integrity="sha256-Vzbj7sDDS/woiFS3uNKo8eIuni59rjyNGtXfstRzStA=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/tomorrow.min.css" crossorigin="anonymous" title="hl-light">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/tomorrow.min.css" crossorigin="anonymous" title="hl-dark" disabled>
      
    

    

    

    
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js" integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin="anonymous" async></script>
      
    
      

      
      

      
    
      

      
      

      
    
      

      
      
        
      

      
    
      

      
      

      
    

  

  
  
  
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lora:400,400i%7CKarla:400,400i,700,700i%7CRoboto+Mono%7CKaushan+Script&display=swap">
  

  
  
  
  
  <link rel="stylesheet" href="/css/academic.css">

  





<script async src="https://www.googletagmanager.com/gtag/js?id=UA-99209585-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];

  function gtag() {
      dataLayer.push(arguments);
  }

  function trackOutboundLink(url, target) {
    gtag('event', 'click', {
         'event_category': 'outbound',
         'event_label': url,
         'transport_type': 'beacon',
         'event_callback': function () {
           if (target !== '_blank') {
             document.location = url;
           }
         }
    });
    console.debug("Outbound link clicked: " + url);
  }

  function onClickCallback(event) {
    if ((event.target.tagName !== 'A') || (event.target.host === window.location.host)) {
      return;
    }
    trackOutboundLink(event.target, event.target.getAttribute('target'));  
  }

  gtag('js', new Date());
  gtag('config', 'UA-99209585-1', {});

  
  document.addEventListener('click', onClickCallback, false);
</script>


  


  
  

  

  <link rel="manifest" href="/index.webmanifest">
  <link rel="icon" type="image/png" href="/images/icon_hubd7880285348c4660acc9402140a2483_242472_32x32_fill_lanczos_center_2.png">
  <link rel="apple-touch-icon" type="image/png" href="/images/icon_hubd7880285348c4660acc9402140a2483_242472_192x192_fill_lanczos_center_2.png">

  <link rel="canonical" href="https://taoyan.netlify.app/post/2019-08-12.rehh%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%80%89%E6%8B%A9%E5%8D%B0%E8%AE%B0%E6%89%AB%E6%8F%8F/">

  
  
  
  
  
  
  
    
  
  
  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@TaoYANLogos">
  <meta property="twitter:creator" content="@TaoYANLogos">
  
  <meta property="og:site_name" content="Tao Yan">
  <meta property="og:url" content="https://taoyan.netlify.app/post/2019-08-12.rehh%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%80%89%E6%8B%A9%E5%8D%B0%E8%AE%B0%E6%89%AB%E6%8F%8F/">
  <meta property="og:title" content="rehh:全基因组选择印记扫描 | Tao Yan">
  <meta property="og:description" content="
简介
Sabeti在2002年的时候提出了一种叫做Extended Haplotype Homozygosity (EHH)的统计方法，用来扫描全基因组范围内的选择印记。在此基础上又衍生出了iHS、XP-EHH以及Rsb等统计方法。R包rehh实现了这几种统计方法的封装。"><meta property="og:image" content="https://taoyan.netlify.app/img/pom-card.png">
  <meta property="twitter:image" content="https://taoyan.netlify.app/img/pom-card.png"><meta property="og:locale" content="en-us">
  
    
      <meta property="article:published_time" content="2019-08-12T18:19:05&#43;00:00">
    
    <meta property="article:modified_time" content="2019-08-12T18:19:05&#43;00:00">
  

  


    






  




<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "https://taoyan.netlify.app/post/2019-08-12.rehh%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%80%89%E6%8B%A9%E5%8D%B0%E8%AE%B0%E6%89%AB%E6%8F%8F/"
  },
  "headline": "rehh:全基因组选择印记扫描",
  
  "datePublished": "2019-08-12T18:19:05Z",
  "dateModified": "2019-08-12T18:19:05Z",
  
  "publisher": {
    "@type": "Organization",
    "name": "Tao Yan",
    "logo": {
      "@type": "ImageObject",
      "url": "https://taoyan.netlify.app/images/icon_hubd7880285348c4660acc9402140a2483_242472_192x192_fill_lanczos_center_2.png"
    }
  },
  "description": "\u003cp\u003e\u003cimg src=\"https://raw.githubusercontent.com/YTLogos/pic_link/master/img/20190814134451.png\" alt=\"\"\u003e\u003c/p\u003e\n\u003ch3 id=\"简介\"\u003e简介\u003c/h3\u003e\n\u003cp\u003eSabeti在2002年的时候提出了一种叫做\u003ccode\u003eExtended Haplotype Homozygosity (EHH)\u003c/code\u003e的统计方法，用来扫描全基因组范围内的选择印记。在此基础上又衍生出了\u003ccode\u003eiHS\u003c/code\u003e、\u003ccode\u003eXP-EHH\u003c/code\u003e以及\u003ccode\u003eRsb\u003c/code\u003e等统计方法。R包\u003ccode\u003erehh\u003c/code\u003e实现了这几种统计方法的封装。\u003c/p\u003e"
}
</script>

  

  


  


  





  <title>rehh:全基因组选择印记扫描 | Tao Yan</title>

</head>

<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" >

  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


  
<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">Tao Yan</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav ml-auto">
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link  active" href="/post"><span>Blog</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/publication"><span>Publications</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/projects"><span>Projects</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/CV.html"><span>CV</span></a>
        </li>

        
        

      

        

        
        <li class="nav-item">
          <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        

        

      </ul>

    </div>
  </div>
</nav>


  <article class="article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1>rehh:全基因组选择印记扫描</h1>

  

  
    



<meta content="2019-08-12 18:19:05 &#43;0000 UTC" itemprop="datePublished">
<meta content="2019-08-12 18:19:05 &#43;0000 UTC" itemprop="dateModified">

<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
    
    <time>Aug 12, 2019</time>
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    6 min read
  </span>
  

  
  

  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fas fa-folder"></i>
    <a href="/categories/genome/">Genome</a></span>
  

  
    
<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=https://taoyan.netlify.app/post/2019-08-12.rehh%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%80%89%E6%8B%A9%E5%8D%B0%E8%AE%B0%E6%89%AB%E6%8F%8F/&amp;text=rehh:%e5%85%a8%e5%9f%ba%e5%9b%a0%e7%bb%84%e9%80%89%e6%8b%a9%e5%8d%b0%e8%ae%b0%e6%89%ab%e6%8f%8f" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=https://taoyan.netlify.app/post/2019-08-12.rehh%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%80%89%E6%8B%A9%E5%8D%B0%E8%AE%B0%E6%89%AB%E6%8F%8F/&amp;t=rehh:%e5%85%a8%e5%9f%ba%e5%9b%a0%e7%bb%84%e9%80%89%e6%8b%a9%e5%8d%b0%e8%ae%b0%e6%89%ab%e6%8f%8f" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=rehh:%e5%85%a8%e5%9f%ba%e5%9b%a0%e7%bb%84%e9%80%89%e6%8b%a9%e5%8d%b0%e8%ae%b0%e6%89%ab%e6%8f%8f&amp;body=https://taoyan.netlify.app/post/2019-08-12.rehh%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%80%89%E6%8B%A9%E5%8D%B0%E8%AE%B0%E6%89%AB%E6%8F%8F/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=https://taoyan.netlify.app/post/2019-08-12.rehh%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%80%89%E6%8B%A9%E5%8D%B0%E8%AE%B0%E6%89%AB%E6%8F%8F/&amp;title=rehh:%e5%85%a8%e5%9f%ba%e5%9b%a0%e7%bb%84%e9%80%89%e6%8b%a9%e5%8d%b0%e8%ae%b0%e6%89%ab%e6%8f%8f" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
  </ul>
</div>


  

</div>
    














  
</div>



  <div class="article-container">

    <div class="article-style">
      <p><img src="https://raw.githubusercontent.com/YTLogos/pic_link/master/img/20190814134451.png" alt=""></p>
<h3 id="简介">简介</h3>
<p>Sabeti在2002年的时候提出了一种叫做<code>Extended Haplotype Homozygosity (EHH)</code>的统计方法，用来扫描全基因组范围内的选择印记。在此基础上又衍生出了<code>iHS</code>、<code>XP-EHH</code>以及<code>Rsb</code>等统计方法。R包<code>rehh</code>实现了这几种统计方法的封装。</p>
<h3 id="应用">应用</h3>
<p><code>rehh</code>支持非常多类型的输入数据，这里我们使用用的较多的数据类型<code>vcf</code>数据，其他的数据类型可以查看
<a href="https://cran.r-project.org/web/packages/rehh/vignettes/rehh.html" target="_blank" rel="noopener"><code>vignette </code></a>。</p>
<h3 id="数据输入">数据输入</h3>
<p>该包提供了一些示例数据，这里我们使用的是提供的bta12_cgu.vcf.gz数据</p>
<pre><code>$ zcat bta12_cgu.vcf.gz|less

##fileformat=VCFv4.2
##reference=Btau20070913-freeze
##INFO=&lt;ID=NS,Number=1,Type=Integer,Description=&quot;Number of Samples With Data&quot;&gt;
##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  CGU_MN026       CGU_MN027
12      79823   F1200140        T       A       .       PASS    NS=140  GT      1|1     0|1     1|1
12      125974  F1200150        C       G       .       PASS    NS=140  GT      1|1     1|0     1|0
12      175087  F1200170        G       C       .       PASS    NS=140  GT      0|0     1|1     1|1
12      219152  F1200180        A       T       .       PASS    NS=140  GT      1|1     1|1     1|1
12      256896  F1200190        T       A       .       PASS    NS=140  GT      0|1     1|1     1|1
12      316254  F1200210        C       G       .       PASS    NS=140  GT      1|0     0|1     0|0
</code></pre>
<p>读取vcf数据之前，先安装R包vcfR</p>
<pre><code>if(!require(vcfR))
  BiocManager::install(&quot;vcfR&quot;)
if(!require(rehh))
  install.packages(&quot;rehh&quot;)

## 数据读取、转换，rehh提供了一个函数data2haplohh

hh &lt;- data2haplohh(hap_file = &quot;bta12_cgu.vcf.gz&quot;,
                   polarize_vcf = FALSE)
* Reading input file(s) *
Scanning file to determine attributes.
File attributes:
  meta lines: 4
  header_line: 5
  variant count: 1424
  column count: 149
Meta line 4 read in.
All meta lines processed.
gt matrix initialized.
Character matrix gt created.
  Character matrix gt rows: 1424
  Character matrix gt cols: 149
  skip: 0
  nrows: 1424
  row_num: 0
Processed variant: 1424
All variants processed
Extracting map information.
Extracting haplotypes.
Number of individuals which are 
Haploid Diploid Triploid, ... : 
1 2 
0 140 
* Filtering data *
Discard markers genotyped on less than 100 % of haplotypes.
No marker discarded.
Data consists of 280 haplotypes and 1424 markers.
Number of mono-, bi-, multi-allelic markers:
1 2 
27 1397                  
</code></pre>
<h3 id="计算ehhs可视化">计算EHHS、可视化</h3>
<pre><code>#这里我们选择SNP位点F1205400
res &lt;- calc_ehhs(hh, 
                 mrk = &quot;F1205400&quot;, 
                 include_nhaplo = TRUE)
#可视化EHHS
plot(res)
</code></pre>
<p><img src="https://raw.githubusercontent.com/YTLogos/pic_link/master/img/20190812142451.png" alt=""></p>
<blockquote>
<p><code>calc_ehhs</code>会计算出<code>EHHS</code>以及<code>nEHHS(un-normalized EHHS)</code>，<code>plot</code>默认可视化的是<code>EHHS</code>，也可以绘制<code>nEHHS</code>，此时<code>focal marker</code>也就是我们这里选的<code>F1205400</code> SNP处的<code>EHHS</code>的值会被归一化为1.</p>
</blockquote>
<pre><code>plot(res, nehhs = TRUE)
</code></pre>
<p><img src="https://raw.githubusercontent.com/YTLogos/pic_link/master/img/20190812143025.png" alt=""></p>
<h3 id="数据实战">数据实战</h3>
<p>下面使用我们自己的数据来分析，前期我们重测序了1000份油菜，分为三类生态型：春型、半冬型、冬型。我们可以分别检测这三个群体某个位点的选择印记。</p>
<pre><code>#首先提取各个群体的SNP，准备需要提取的样品名称,这里提取S、SW、W
$ less S.txt
R4168
R4190
R4205
R4210
R4216
...

#提取SNP
/public/home/jianglix/biosoft/vcftools/bin/vcftools --gzvcf Bna.snp.maf.0.05.int.0.5.vcf.gz --keep ./s --recode --recode-INFO-all -c &gt; pop-S.vcf
/public/home/jianglix/biosoft/vcftools/bin/vcftools --gzvcf Bna.snp.maf.0.05.int.0.5.vcf.gz --keep ./w --recode --recode-INFO-all -c &gt; pop-W.vcf
/public/home/jianglix/biosoft/vcftools/bin/vcftools --gzvcf Bna.snp.maf.0.05.int.0.5.vcf.gz --keep ./sw --recode --recode-INFO-all -c &gt; pop-SW.vcf

#建立tabix索引
~/biosoft/htslib1.9/bin/bgzip &lt; pop-S.vcf &gt; pop-S.vcf.gz
~/biosoft/htslib1.9/bin/tabix -p vcf pop-S.vcf.gz

~/biosoft/htslib1.9/bin/bgzip &lt; pop-SW.vcf &gt; pop-SW.vcf.gz
~/biosoft/htslib1.9/bin/tabix -p vcf pop-SW.vcf.gz

~/biosoft/htslib1.9/bin/bgzip &lt; pop-W.vcf &gt; pop-W.vcf.gz
~/biosoft/htslib1.9/bin/tabix -p vcf pop-W.vcf.gz

##如果SNP缺失过多，可以基因型填补，因为缺失太多会影响后续分析
java -jar -Xmx110G &quot;-Djava.io.tmpdir=/database/tmp/&quot; ~/biosoft/beagle/beagle5.jar gt=pop-S.vcf.gz out=pop-S.imputed
java -jar -Xmx110G &quot;-Djava.io.tmpdir=/database/tmp/&quot; ~/biosoft/beagle/beagle5.jar gt=pop-SW.vcf.gz out=pop-SW.imputed
java -jar -Xmx110G &quot;-Djava.io.tmpdir=/database/tmp/&quot; ~/biosoft/beagle/beagle5.jar gt=pop-W.vcf.gz out=pop-W.imputed
</code></pre>
<h3 id="基因选取">基因选取</h3>
<p>我们可以选择某些基因区域来查看此基因在不同的群体（春、冬、半冬）中是否受到选择，这里我们选择开花相关基因<em>FLC.A10</em> (20022944~20027730)。选取基因上下游50KB进行分析</p>
<pre><code>#bed文件，用于提取SNP
10 19982944 20067730

#SNP提取
~/biosoft/htslib1.9/bin/tabix -R flc.A10.bed -h pop-S-imputed.vcf.gz &gt; pop-S.flc.A10.vcf
~/biosoft/htslib1.9/bin/tabix -R flc.A10.bed -h pop-SW-imputed.vcf.gz &gt; pop-SW.flc.A10.vcf
~/biosoft/htslib1.9/bin/tabix -R flc.A10.bed -h pop-W-imputed.vcf.gz&gt; pop-W.flc.A10.vcf
</code></pre>
<h3 id="ehhs分析">EHHS分析</h3>
<p>一般来说是根据<code>GWAS</code>或者其它选择性清除分析结果来选取<code>focal marker</code>,这里我就随便选个<code>SNP(10_20025969)</code></p>
<pre><code>library(rehh)
library(tidyverse)

s &lt;- data2haplohh(&quot;pop-S.flc.A10.vcf&quot;, polarize_vcf = FALSE)
sw &lt;- data2haplohh(&quot;pop-SW.flc.A10.vcf&quot;, polarize_vcf = FALSE)
w &lt;- data2haplohh(&quot;pop-W.flc.A10.vcf&quot;, polarize_vcf = FALSE)

snp &lt;- &quot;10_20025969&quot;

res_s &lt;- calc_ehhs(
  s,
  mrk = snp,
  include_nhaplo = TRUE
  )

res_sw &lt;- calc_ehhs(
  sw,
  mrk = snp,
  include_nhaplo = TRUE
  )

res_w &lt;- calc_ehhs(
  w,
  mrk = snp,
  include_nhaplo = TRUE
  )

data_s &lt;- res_s$ehhs %&gt;%
  mutate(pos = POSITION / 1000000) %&gt;%
  mutate(type = &quot;Spring&quot;)

data_sw &lt;- res_sw$ehhs %&gt;%
  mutate(pos = POSITION / 1000000) %&gt;%
  mutate(type = &quot;Semi-Winter&quot;)

data_w &lt;- res_w$ehhs %&gt;%
  mutate(pos = POSITION / 1000000) %&gt;%
  mutate(type = &quot;Winter&quot;)

data_all &lt;- rbind(data_s, data_sw, data_w)

p &lt;- ggplot(data_all, aes(pos,NEHHS,color=type))+
  geom_line(size=1.25)+
  theme_classic()+
  theme(legend.title = element_blank(), legend.position = c(0.75,0.85))+
  scale_color_manual(values = c(&quot;#22ac38&quot;,&quot;#6299fd&quot;,&quot;#63c2cb&quot;))+
  labs(x=&quot;Chromosome A10 (Mb)&quot;, y=&quot;EHHS&quot;)

print(p)
</code></pre>
<p><img src="https://raw.githubusercontent.com/YTLogos/pic_link/master/img/20190814134451.png" alt=""></p>
<p>可以看出差异不大，春冬半冬之间的受选择差异不大，这有可能跟挑选的位点有关。</p>
<h2 id="sessioninfo">SessionInfo</h2>
<pre><code>devtools::session_info()
- Session info -------------------------------------------------------------------------------------------------------------------------------------------------
 setting  value                         
 version  R version 3.6.1 (2019-07-05)  
 os       Windows 10 x64                
 system   x86_64, mingw32               
 ui       RStudio                       
 language (EN)                          
 collate  Chinese (Simplified)_China.936
 ctype    Chinese (Simplified)_China.936
 tz       Asia/Taipei                   
 date     2019-08-14                    

- Packages -----------------------------------------------------------------------------------------------------------------------------------------------------
 package     * version  date       lib source                             
 ape           5.3      2019-03-17 [1] CRAN (R 3.6.0)                     
 assertthat    0.2.1    2019-03-21 [1] CRAN (R 3.6.0)                     
 backports     1.1.4    2019-04-10 [1] CRAN (R 3.6.0)                     
 broom         0.5.2    2019-04-07 [1] CRAN (R 3.6.0)                     
 callr         3.3.1    2019-07-18 [1] CRAN (R 3.6.1)                     
 cellranger    1.1.0    2016-07-27 [1] CRAN (R 3.6.0)                     
 cli           1.1.0    2019-03-19 [1] CRAN (R 3.6.0)                     
 cluster       2.1.0    2019-06-19 [2] CRAN (R 3.6.1)                     
 colorspace    1.4-1    2019-03-18 [1] CRAN (R 3.6.0)                     
 crayon        1.3.4    2017-09-16 [1] CRAN (R 3.6.0)                     
 desc          1.2.0    2018-05-01 [1] CRAN (R 3.6.0)                     
 devtools      2.1.0    2019-07-06 [1] CRAN (R 3.6.1)                     
 digest        0.6.20   2019-07-04 [1] CRAN (R 3.6.1)                     
 dplyr       * 0.8.3    2019-07-04 [1] CRAN (R 3.6.1)                     
 fansi         0.4.0    2018-10-05 [1] CRAN (R 3.6.0)                     
 forcats     * 0.4.0    2019-02-17 [1] CRAN (R 3.6.0)                     
 fs            1.3.1    2019-05-06 [1] CRAN (R 3.6.0)                     
 gapminder   * 0.3.0    2017-10-31 [2] CRAN (R 3.6.0)                     
 generics      0.0.2    2018-11-29 [2] CRAN (R 3.6.0)                     
 ggplot2     * 3.2.1    2019-08-10 [1] CRAN (R 3.6.1)                     
 glue          1.3.1    2019-03-12 [1] CRAN (R 3.6.0)                     
 grkstyle      0.0.1    2019-08-13 [1] Github (gadenbuie/grkstyle@a141d39)
 gtable        0.3.0    2019-03-25 [1] CRAN (R 3.6.0)                     
 haven         2.1.1    2019-07-04 [1] CRAN (R 3.6.1)                     
 hms           0.5.0    2019-07-09 [1] CRAN (R 3.6.1)                     
 httr          1.4.1    2019-08-05 [1] CRAN (R 3.6.1)                     
 jsonlite      1.6      2018-12-07 [1] CRAN (R 3.6.0)                     
 labeling      0.3      2014-08-23 [1] CRAN (R 3.6.0)                     
 lattice       0.20-38  2018-11-04 [2] CRAN (R 3.6.1)                     
 lazyeval      0.2.2    2019-03-15 [1] CRAN (R 3.6.0)                     
 lubridate     1.7.4    2018-04-11 [1] CRAN (R 3.6.0)                     
 magrittr      1.5      2014-11-22 [1] CRAN (R 3.6.0)                     
 MASS          7.3-51.4 2019-03-31 [2] CRAN (R 3.6.1)                     
 Matrix        1.2-17   2019-03-22 [1] CRAN (R 3.6.0)                     
 memoise       1.1.0    2017-04-21 [1] CRAN (R 3.6.0)                     
 memuse        4.0-0    2017-11-10 [1] CRAN (R 3.6.0)                     
 mgcv          1.8-28   2019-03-21 [2] CRAN (R 3.6.1)                     
 modelr        0.1.5    2019-08-08 [1] CRAN (R 3.6.1)                     
 munsell       0.5.0    2018-06-12 [1] CRAN (R 3.6.0)                     
 nlme          3.1-141  2019-08-01 [2] CRAN (R 3.6.1)                     
 permute       0.9-5    2019-03-12 [1] CRAN (R 3.6.0)                     
 pillar        1.4.2    2019-06-29 [1] CRAN (R 3.6.0)                     
 pinfsc50      1.1.0    2016-12-02 [1] CRAN (R 3.6.0)                     
 pkgbuild      1.0.4    2019-08-05 [1] CRAN (R 3.6.1)                     
 pkgconfig     2.0.2    2018-08-16 [1] CRAN (R 3.6.0)                     
 pkgload       1.0.2    2018-10-29 [2] CRAN (R 3.6.0)                     
 prettyunits   1.0.2    2015-07-13 [1] CRAN (R 3.6.0)                     
 processx      3.4.1    2019-07-18 [1] CRAN (R 3.6.1)                     
 ps            1.3.0    2018-12-21 [2] CRAN (R 3.6.0)                     
 purrr       * 0.3.2    2019-03-15 [1] CRAN (R 3.6.0)                     
 R6            2.4.0    2019-02-14 [1] CRAN (R 3.6.0)                     
 Rcpp          1.0.2    2019-07-25 [1] CRAN (R 3.6.1)                     
 readr       * 1.3.1    2018-12-21 [1] CRAN (R 3.6.0)                     
 readxl        1.3.1    2019-03-13 [1] CRAN (R 3.6.0)                     
 rehh        * 3.0.1    2019-07-11 [1] CRAN (R 3.6.1)                     
 rehh.data     1.0.0    2016-11-08 [1] CRAN (R 3.6.0)                     
 remotes       2.1.0    2019-06-24 [2] CRAN (R 3.6.0)                     
 rlang         0.4.0    2019-06-25 [1] CRAN (R 3.6.0)                     
 rprojroot     1.3-2    2018-01-03 [1] CRAN (R 3.6.0)                     
 rstudioapi    0.10     2019-03-19 [1] CRAN (R 3.6.0)                     
 rvest         0.3.4    2019-05-15 [1] CRAN (R 3.6.0)                     
 scales        1.0.0    2018-08-09 [1] CRAN (R 3.6.0)                     
 sessioninfo   1.1.1    2018-11-05 [2] CRAN (R 3.6.0)                     
 stringi       1.4.3    2019-03-12 [1] CRAN (R 3.6.0)                     
 stringr     * 1.4.0    2019-02-10 [1] CRAN (R 3.6.0)                     
 styler        1.1.1    2019-05-06 [1] CRAN (R 3.6.0)                     
 testthat      2.2.1    2019-07-25 [1] CRAN (R 3.6.1)                     
 tibble      * 2.1.3    2019-06-06 [1] CRAN (R 3.6.0)                     
 tidyr       * 0.8.3    2019-03-01 [1] CRAN (R 3.6.0)                     
 tidyselect    0.2.5    2018-10-11 [1] CRAN (R 3.6.0)                     
 tidyverse   * 1.2.1    2017-11-14 [1] CRAN (R 3.6.0)                     
 usethis       1.5.1    2019-07-04 [1] CRAN (R 3.6.1)                     
 utf8          1.1.4    2018-05-24 [1] CRAN (R 3.6.0)                     
 vcfR          1.8.0    2018-04-17 [1] CRAN (R 3.6.0)                     
 vctrs         0.2.0    2019-07-05 [1] CRAN (R 3.6.1)                     
 vegan         2.5-5    2019-05-12 [1] CRAN (R 3.6.0)                     
 viridisLite   0.3.0    2018-02-01 [1] CRAN (R 3.6.0)                     
 withr         2.1.2    2018-03-15 [1] CRAN (R 3.6.0)                     
 xml2          1.2.2    2019-08-09 [1] CRAN (R 3.6.1)                     
 zeallot       0.1.0    2018-01-28 [1] CRAN (R 3.6.0)                     

[1] C:/Tools/R/R_Library
[2] C:/Tools/R-3.6.1/library
</code></pre>
    </div>

    






<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/selection/">selection</a>
  
  <a class="badge badge-light" href="/tags/domestication/">domestication</a>
  
</div>



<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=https://taoyan.netlify.app/post/2019-08-12.rehh%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%80%89%E6%8B%A9%E5%8D%B0%E8%AE%B0%E6%89%AB%E6%8F%8F/&amp;text=rehh:%e5%85%a8%e5%9f%ba%e5%9b%a0%e7%bb%84%e9%80%89%e6%8b%a9%e5%8d%b0%e8%ae%b0%e6%89%ab%e6%8f%8f" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=https://taoyan.netlify.app/post/2019-08-12.rehh%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%80%89%E6%8B%A9%E5%8D%B0%E8%AE%B0%E6%89%AB%E6%8F%8F/&amp;t=rehh:%e5%85%a8%e5%9f%ba%e5%9b%a0%e7%bb%84%e9%80%89%e6%8b%a9%e5%8d%b0%e8%ae%b0%e6%89%ab%e6%8f%8f" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=rehh:%e5%85%a8%e5%9f%ba%e5%9b%a0%e7%bb%84%e9%80%89%e6%8b%a9%e5%8d%b0%e8%ae%b0%e6%89%ab%e6%8f%8f&amp;body=https://taoyan.netlify.app/post/2019-08-12.rehh%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%80%89%E6%8B%A9%E5%8D%B0%E8%AE%B0%E6%89%AB%E6%8F%8F/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=https://taoyan.netlify.app/post/2019-08-12.rehh%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%80%89%E6%8B%A9%E5%8D%B0%E8%AE%B0%E6%89%AB%E6%8F%8F/&amp;title=rehh:%e5%85%a8%e5%9f%ba%e5%9b%a0%e7%bb%84%e9%80%89%e6%8b%a9%e5%8d%b0%e8%ae%b0%e6%89%ab%e6%8f%8f" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
  </ul>
</div>












  
  





  
    
    
    
      
    
    
    
    <div class="media author-card content-widget-hr">
      
        
        <img class="avatar mr-3 avatar-circle" src="/authors/alison/avatar_hufa245e75a5e5fa47274790393ba89624_26717_270x270_fill_q90_lanczos_center.jpg" alt="">
      

      <div class="media-body">
        <h5 class="card-title"><a href="https://taoyan.netlify.app/"></a></h5>
        <h6 class="card-subtitle">PhD Candidate of Crop Genetics and Breeding</h6>
        <p class="card-text">I am a PhD student of Crop Genetics and Breeding at the <em>Zhejiang University</em> Crop Science Lab. My research interests covers a range of issues:Population Genetics Evolution and Ecotype Divergence Analysis of Oilseed Rape, Genome-wide Association Study (GWAS) of Agronomic Traits.</p>
        <ul class="network-icon" aria-hidden="true">
  
    
    
    
      
    
    
    
    
    
    <li>
      <a href="mailto:tyan@zju.edu.cn" >
        <i class="fas fa-paper-plane"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
    <li>
      <a href="mailto:tyan@zju.edu.cn" >
        <i class="fas fa-envelope"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://twitter.com/TaoYANLogos" target="_blank" rel="noopener">
        <i class="fab fa-twitter"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://github.com/YTLogos" target="_blank" rel="noopener">
        <i class="fab fa-github"></i>
      </a>
    </li>
  
    
    
    
    
    
    
    
      
    
    <li>
      <a href="https://scholar.google.com/citations?hl=en&amp;user=8p3wQNwAAAAJ" target="_blank" rel="noopener">
        <i class="ai ai-google-scholar"></i>
      </a>
    </li>
  
</ul>

      </div>
    </div>
  







<section id="comments">
  
    
<div id="disqus_thread"></div>
<script>
  let disqus_config = function () {
    
    
    
  };
  (function() {
    if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
      document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
      return;
    }
    var d = document, s = d.createElement('script'); s.async = true;
    s.src = 'https://' + "logos-1" + '.disqus.com/embed.js';
    s.setAttribute('data-timestamp', +new Date());
    (d.head || d.body).appendChild(s);
  })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>


  
</section>










  </div>
</article>

      

    
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js" integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin="anonymous"></script>

      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js" integrity="sha256-eOgo0OtLL4cdq7RdwRUiGKLX9XsIJ7nGhWEKbohmVAQ=" crossorigin="anonymous"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/r.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/Shell.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/Python.min.js"></script>
        
      

      
      
    

    
    

    
    
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    
    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.37431be2d92d7fb0160054761ab79602.js"></script>

    






  
  
  <div class="container">
    <footer class="site-footer">
  
  <p class="powered-by">
    <a href="/license/">LICENSE: CC-BY-SA <br> <i class="fab fa-creative-commons fa-2x"></i><i class="fab fa-creative-commons-by fa-2x"></i><i class="fab fa-creative-commons-sa fa-2x"></i> </a><br>
  </p>
  

  <p class="powered-by">
    © Tao Yan, 2018-2020 &middot; 

    Made with <i class="far fa-heart" style="color:#E13D3D"></i>, <a href="https://github.com/YTLogos/TaoYan" target="_blank" rel="noopener"><i class="fas fa-code-branch" style="color:#E13D3D"></i></a>, the <a href="https://cran.r-project.org/" target="_blank" rel="noopener"><i class="fab fa-r-project"></i> </a><a href="https://github.com/rstudio/blogdown" target="_blank" rel="noopener">blogdown</a> package, and the 
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" id="back_to_top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

<div style="text-align: center;"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/6.js?i=5cz8zk3uynm&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=0" async="async"></script></div>

  </div>
  

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

</body>
</html>
