<!DOCTYPE html><html lang="en-us" >

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 4.8.0">

  

  
  
  
  
  
    
    
    
  
  

  

  
  
  
    
  
  <meta name="description" content="
最近有一批转录组数据需要分析，从公司拿到raw data之后进行后续分析。目前转录组测序十分成熟，测序质量都很高，所以我这里就简单地质控一下，由
fastp完成。">

  
  <link rel="alternate" hreflang="en-us" href="https://taoyan.netlify.app/post/2019-09-03.%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%9001/">

  


  
  
  
  <meta name="theme-color" content="#06805E">
  

  
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.0-1/css/all.min.css" integrity="sha256-4w9DunooKSr3MFXHXWyFER38WmPdm361bQS/2KUWZbU=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css" integrity="sha256-Vzbj7sDDS/woiFS3uNKo8eIuni59rjyNGtXfstRzStA=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/tomorrow.min.css" crossorigin="anonymous" title="hl-light">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/tomorrow.min.css" crossorigin="anonymous" title="hl-dark" disabled>
      
    

    

    

    
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js" integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin="anonymous" async></script>
      
    
      

      
      

      
    
      

      
      

      
    
      

      
      
        
      

      
    
      

      
      

      
    

  

  
  
  
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lora:400,400i%7CKarla:400,400i,700,700i%7CRoboto+Mono%7CKaushan+Script&display=swap">
  

  
  
  
  
  <link rel="stylesheet" href="/css/academic.css">

  





<script async src="https://www.googletagmanager.com/gtag/js?id=UA-99209585-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];

  function gtag() {
      dataLayer.push(arguments);
  }

  function trackOutboundLink(url, target) {
    gtag('event', 'click', {
         'event_category': 'outbound',
         'event_label': url,
         'transport_type': 'beacon',
         'event_callback': function () {
           if (target !== '_blank') {
             document.location = url;
           }
         }
    });
    console.debug("Outbound link clicked: " + url);
  }

  function onClickCallback(event) {
    if ((event.target.tagName !== 'A') || (event.target.host === window.location.host)) {
      return;
    }
    trackOutboundLink(event.target, event.target.getAttribute('target'));  
  }

  gtag('js', new Date());
  gtag('config', 'UA-99209585-1', {});

  
  document.addEventListener('click', onClickCallback, false);
</script>


  


  
  

  

  <link rel="manifest" href="/index.webmanifest">
  <link rel="icon" type="image/png" href="/images/icon_hubd7880285348c4660acc9402140a2483_242472_32x32_fill_lanczos_center_2.png">
  <link rel="apple-touch-icon" type="image/png" href="/images/icon_hubd7880285348c4660acc9402140a2483_242472_192x192_fill_lanczos_center_2.png">

  <link rel="canonical" href="https://taoyan.netlify.app/post/2019-09-03.%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%9001/">

  
  
  
  
  
  
  
    
  
  
  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@TaoYANLogos">
  <meta property="twitter:creator" content="@TaoYANLogos">
  
  <meta property="og:site_name" content="Tao Yan">
  <meta property="og:url" content="https://taoyan.netlify.app/post/2019-09-03.%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%9001/">
  <meta property="og:title" content="转录组数据分析01-比对 | Tao Yan">
  <meta property="og:description" content="
最近有一批转录组数据需要分析，从公司拿到raw data之后进行后续分析。目前转录组测序十分成熟，测序质量都很高，所以我这里就简单地质控一下，由
fastp完成。"><meta property="og:image" content="https://taoyan.netlify.app/img/pom-card.png">
  <meta property="twitter:image" content="https://taoyan.netlify.app/img/pom-card.png"><meta property="og:locale" content="en-us">
  
    
      <meta property="article:published_time" content="2019-09-03T16:19:05&#43;00:00">
    
    <meta property="article:modified_time" content="2019-09-03T16:19:05&#43;00:00">
  

  


    






  




<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "https://taoyan.netlify.app/post/2019-09-03.%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%9001/"
  },
  "headline": "转录组数据分析01-比对",
  
  "datePublished": "2019-09-03T16:19:05Z",
  "dateModified": "2019-09-03T16:19:05Z",
  
  "publisher": {
    "@type": "Organization",
    "name": "Tao Yan",
    "logo": {
      "@type": "ImageObject",
      "url": "https://taoyan.netlify.app/images/icon_hubd7880285348c4660acc9402140a2483_242472_192x192_fill_lanczos_center_2.png"
    }
  },
  "description": "\u003cp\u003e\u003cimg src=\"https://raw.githubusercontent.com/YTLogos/pic_link/master/img/20190904101824.png\" alt=\"\"\u003e\u003c/p\u003e\n\u003cp\u003e最近有一批转录组数据需要分析，从公司拿到\u003ccode\u003eraw data\u003c/code\u003e之后进行后续分析。目前转录组测序十分成熟，测序质量都很高，所以我这里就简单地质控一下，由\n\u003ca href=\"https://github.com/OpenGene/fastp\" target=\"_blank\" rel=\"noopener\"\u003e\u003cstrong\u003efastp\u003c/strong\u003e\u003c/a\u003e完成。\u003c/p\u003e"
}
</script>

  

  


  


  





  <title>转录组数据分析01-比对 | Tao Yan</title>

</head>

<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" >

  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


  
<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">Tao Yan</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav ml-auto">
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link  active" href="/post"><span>Blog</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/publication"><span>Publications</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/projects"><span>Projects</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/CV.html"><span>CV</span></a>
        </li>

        
        

      

        

        
        <li class="nav-item">
          <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        

        

      </ul>

    </div>
  </div>
</nav>


  <article class="article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1>转录组数据分析01-比对</h1>

  

  
    



<meta content="2019-09-03 16:19:05 &#43;0000 UTC" itemprop="datePublished">
<meta content="2019-09-03 16:19:05 &#43;0000 UTC" itemprop="dateModified">

<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
    
    <time>Sep 3, 2019</time>
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    3 min read
  </span>
  

  
  

  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fas fa-folder"></i>
    <a href="/categories/rna-seq/">RNA-seq</a></span>
  

  
    
<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=https://taoyan.netlify.app/post/2019-09-03.%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%9001/&amp;text=%e8%bd%ac%e5%bd%95%e7%bb%84%e6%95%b0%e6%8d%ae%e5%88%86%e6%9e%9001-%e6%af%94%e5%af%b9" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=https://taoyan.netlify.app/post/2019-09-03.%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%9001/&amp;t=%e8%bd%ac%e5%bd%95%e7%bb%84%e6%95%b0%e6%8d%ae%e5%88%86%e6%9e%9001-%e6%af%94%e5%af%b9" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=%e8%bd%ac%e5%bd%95%e7%bb%84%e6%95%b0%e6%8d%ae%e5%88%86%e6%9e%9001-%e6%af%94%e5%af%b9&amp;body=https://taoyan.netlify.app/post/2019-09-03.%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%9001/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=https://taoyan.netlify.app/post/2019-09-03.%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%9001/&amp;title=%e8%bd%ac%e5%bd%95%e7%bb%84%e6%95%b0%e6%8d%ae%e5%88%86%e6%9e%9001-%e6%af%94%e5%af%b9" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
  </ul>
</div>


  

</div>
    














  
</div>



  <div class="article-container">

    <div class="article-style">
      <p><img src="https://raw.githubusercontent.com/YTLogos/pic_link/master/img/20190904101824.png" alt=""></p>
<p>最近有一批转录组数据需要分析，从公司拿到<code>raw data</code>之后进行后续分析。目前转录组测序十分成熟，测序质量都很高，所以我这里就简单地质控一下，由
<a href="https://github.com/OpenGene/fastp" target="_blank" rel="noopener"><strong>fastp</strong></a>完成。</p>
<h2 id="质控">质控</h2>
<pre><code>#安装fastp
# get source (you can also use browser to download from master or releases)
git clone https://github.com/OpenGene/fastp.git

# build
cd fastp
make

# Install
sudo make install

#没有root权限的话可以使用conda安装或者下载二进制文件
# note: the fastp version in bioconda may be not the latest
conda install -c bioconda fastp

# this binary was compiled on CentOS, and tested on CentOS/Ubuntu
wget http://opengene.org/fastp/fastp
chmod a+x ./fastp
</code></pre>
<p>之后批量进行质控，一条龙服务，十分方便了。</p>
<pre><code>#!/bin/bash
for i in {01..45}
do
~/biosoft/fastp/bin/fastp -i /database/Bna-rna-seq-analyis/data/Cole_M838-01-T${i}_good_1.fq.gz -o /database/Bna-rna-seq-analyis/data/T${i}_good_1.fq.gz -I /database/Bna-rna-seq-analyis/data/Cole_M838-01-T${i}_good_2.fq.gz -O /database/Bna-rna-seq-analyis/data/T${i}_good_2.fq.gz
done
</code></pre>
<h2 id="比对">比对</h2>
<p>比对软件有很多，主流的主要有
<a href="http://ccb.jhu.edu/software/hisat2/index.shtml" target="_blank" rel="noopener">HISAT2</a>以及
<a href="https://github.com/alexdobin/STAR" target="_blank" rel="noopener">STAR</a>。</p>
<h3 id="star比对">STAR比对</h3>
<pre><code>#Download the latest release from and uncompress it
# Get latest STAR source from releases
wget https://github.com/alexdobin/STAR/archive/2.7.2b.tar.gz
tar -xzf 2.7.2b.tar.gz
cd STAR-2.7.2b

# Alternatively, get STAR source using git
git clone https://github.com/alexdobin/STAR.git

#Compile under Linux
# Compile
cd STAR/source
make STAR
</code></pre>
<p><code>STAR</code>使用格式如下：</p>
<pre><code>STAR --option1-name option1-value --option2-name option2-value ...
</code></pre>
<p><code>STAR</code>的文档写的非常详细，这也是一个好软件的标志之一，如果一个软件的文档都写的不详细的话，可以考虑换一个软件。<code>STAR</code>参数非常多，尤其现在还支持三代测序数据。有兴趣的话可以自己去研究
<a href="https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf" target="_blank" rel="noopener">文档</a>。</p>
<h4 id="star构建索引">STAR构建索引</h4>
<pre><code>#!/bin/bash

set -e
set -u
set -o pipefail


fasta=/database/reference/Bna_NY7/NY7-chr/NY_V2_Chr.fasta
gtf=/database/reference/Bna_NY7/NY7-chr/NY7_V2_Chr.gtf
STAR=/home/taoyan/biosoft/STAR/bin/Linux_x86_64_static/STAR

# Generate index Of Brassica napus

$STAR --runMode genomeGenerate \
--runThreadN 32 \
--genomeFastaFiles $fasta \
--genomeDir /database/reference/Bna_NY7/NY7-chr/NY7.Star.Index \
--sjdbGTFfile $gtf \
--genomeSAindexNbases 13 \
--sjdbOverhang 149
</code></pre>
<p>构建索引常用命令参数如下：</p>
<ul>
<li>&ndash;runMode genomeGenerate：运行模式为构建索引</li>
<li>&ndash;runThreadN：设置线程数</li>
<li>&ndash;genomeDir：构建的索引文件存放路径，需要提前创建该文件夹</li>
<li>&ndash;genomeFastaFiles：参考基因组文件</li>
<li>&ndash;sjdbGTFfile：注释文件gtf，推荐在这一步添加，虽然可以在后续比对的时候添加</li>
<li>&ndash;genomeSAindexNbases 对于非常小的参考基因组，此参数需要按比例缩小，其计算方式为min(14, log2(GenomeLength)/2 - 1)，所以一般为14，但是小基因组比如1M的话，此值应为9，这里按照我的参考基因组大小，应该为14，但是我运行的时候提示我最优值为13，暂时不清楚为何提示我这个值。此值一般在10~15之间，越大需要越多的内存，相应的话速度也越快。</li>
<li>&ndash;sjdbOverhang：读段长度-1，默认为99，我的序列是PE150，所以此值应为149，文档显示99可以达到最理想的效果，所以不知道具体的建库方式的话，此参数设为99就行了。</li>
</ul>
<h4 id="批量比对">批量比对</h4>
<pre><code>#!/bin/bash

set -e
set -u
set -o pipefail


fasta=/database/reference/Bna_NY7/NY7-chr/NY_V2_Chr.fasta
gtf=/database/reference/Bna_NY7/NY7-chr/NY7_V2_Chr.gtf
STAR=/home/taoyan/biosoft/STAR/bin/Linux_x86_64_static/STAR

seq_data=/database/Bna.rnaseq.NY7.analysis/data/seq.data
result=/database/Bna.rnaseq.NY7.analysis/result/Star/Star_map_bam

# sequence mapping use STAR

for i in {01..45}
do
echo &quot; ******* Process Sample T${i} ******* &quot;
$STAR --runThreadN 64 \        #线程数
--genomeDir /database/reference/Bna_NY7/NY7-chr/NY7.Star.Index/ \  #索引目录
--readFilesCommand zcat \   #执行解压缩，输入数据格式为gz文件
--readFilesIn $seq_data/T${i}_good_1.fq.gz \  #输入文件
$seq_data/T${i}_good_2.fq.gz \
--outSAMtype BAM SortedByCoordinate \  #指定输出bam文件并排序
--outFileNamePrefix $result/Bna_Star_T${i}. \  #输出文件路径以及前缀
--outBAMsortingThreadN 20 \   #排序线程数
--quantMode TranscriptomeSAM GeneCounts #将上面基因组比对的信息转化为转录本比对的信息，quantMode TranscriptomeSAM will output alignments translated into trancript coordinates,为了后续使用RSEM定量准备输入文件
done
</code></pre>
<p>一些常用参数解释：</p>
<ul>
<li>&ndash;runThreadN：线程数</li>
<li>&ndash;runMode alignReads : 默认就是比对模式，所以这里我没填写</li>
<li>&ndash;genomeDir：索引文件所在文件夹，就是上面建立索引时的文件夹</li>
<li>&ndash;readFilesCommand zcat：由于现在大部分序列数据都是压缩形式的，所以需要解压缩，我的数据是gz格式的，所以这里是zcat</li>
<li>&ndash;readFilesIn：序列数据</li>
<li>&ndash;outSAMtype BAM SortedByCoordinate：输出文件为bam文件，并且进行了排序，这里应该是按照pos排序的。</li>
<li>&ndash;outBAMsortingThreadN：SAM文件排序生成BAM文件时的线程数，不要设置太大，不然会报错的。</li>
<li>&ndash;quantMode TranscriptomeSAM GeneCounts：这个参数需要特别注意，如果你后续要使用RSEM进行定量，就需要添加TranscriptomeSAM，不然生成的BAM文件RSEM无法使用的，GeneCounts生成用于后续featureCounts分析的输入文件</li>
</ul>
<blockquote>
<p>STAR还有其它很多参数，讲不完的，基本来说一般的转录组分析上面这些参数够用了。</p>
</blockquote>
<blockquote>
<p>由于后续定量我们会用到RSEM，所以这里特别注意需要加参数<code>--quantMode TranscriptomeSAM GeneCounts</code>，这样生成的bam文件可用于后续的RSEM定量以及featureCounts计数。</p>
</blockquote>
<blockquote>
<p>这里比对的时候我遇到了我问题:</p>
</blockquote>
<pre><code>&quot;FATAL ERROR: could not create output file ... Solution: check that you have permission to write this file&quot;
</code></pre>
<blockquote>
<p>Google搜索之后发现是系统限制了最大文件打开的数目，默认是1024，我改为了10000，后续比对正常运行。</p>
</blockquote>
<pre><code>#先查看你的系统打开文件限制数目
$ ulimit -n
1024
#设置一个更大的数目
$ ulimit -n 10000
</code></pre>
<h3 id="hisat2比对">HISAT2比对</h3>
<pre><code>#下载源文件
wget http://ccb.jhu.edu/software/hisat2/downloads/hisat2-2.0.0-beta-source.zip
unzip hisat2-2.0.0-beta-source.zip
cd hisat2-2.0.0-beta-source
make
</code></pre>
<p><code>HISAT2</code>用法如下：</p>
<pre><code>hisat2 [options]* -x &lt;hisat2-idx&gt; {-1 &lt;m1&gt; -2 &lt;m2&gt; | -U &lt;r&gt; | --sra-acc &lt;SRA accession number&gt;} [-S &lt;hit&gt;]
</code></pre>
<p>更详细用法见
<a href="http://ccb.jhu.edu/software/hisat2/manual.shtml" target="_blank" rel="noopener">文档</a></p>
<h4 id="hisat2建立索引">HISAT2建立索引</h4>
<p>我这里遇到一个问题，建立索引之后比对的时候报错：</p>
<pre><code>(ERR): hisat2-align died with signal 11 (SEGV) (core dumped)
</code></pre>
<p>Google发现是建立的索引出问题了，HISAT2建立索引的时候会根据基因组大小来建立小索引还是大索引，一般4G以下的基因组都会默认建立小基因组，32位的，以<code>.ht2</code>为后缀的，我这里报错了，所以我特意设置参数建立了64位的以<code>.ht21</code>为后缀的大索引，后续比对就没问题了</p>
<pre><code>#!/bin/bash

set -e
set -u
set -o pipefail


##构建exons以及splice_sites文件，由于我的注释文件是GFF3文件,所以需要转化成GTF

~/biosoft/gffread/gffread NY7_V2_Chr.gff3 -T -o NY7_V2_Chr.gtf
~/biosoft/hisat2-2.0.0-beta/extract_exons.py NY7_V2_Chr.gtf &gt; NY7_V2_Chr.exons
~/biosoft/hisat2-2.0.0-beta/extract_splice_sites.py NY7_V2_Chr.gtf &gt; NY7_V2_Chr.ss

##构建索引,这里需要提供exons，splite_sites文件，参考基因组文件以及输出文件前缀
##Hisat2-build可以构建任意大小的参考基因组索引，小于4G的参考基因组，构建出来的索引以.ht2为后缀的文件，更大的参考基因组是以.ht21为后缀的文件

~/biosoft/hisat2-2.0.0-beta/hisat2-build -p 64 --large-index --ss NY7_V2_Chr.ss --exon NY7_V2_Chr.exon NY_V2_Chr.fasta NY7.Hisat2.Index/NY7_V2_Chr_Hisat2_index 2 &gt; Hisat_build_index.log
</code></pre>
<h4 id="批量比对-1">批量比对</h4>
<pre><code>#!/bin/bash

set -e
set -u
set -o pipefail

hisat2=/home/taoyan/biosoft/hisat2-2.0.0-beta/hisat2
index=/database/reference/Bna_NY7/NY7-chr/NY7.Hisat2.Index
seq_data=/database/Bna.rnaseq.NY7.analysis/data/seq.data
result=/database/Bna.rnaseq.NY7.analysis/result/Hisat2
samtools=/home/taoyan/biosoft/samtools1.9/bin/samtools

for i in {01..45}
do
$hisat2 -t -p 64 --dta -x $index/NY7_V2_Chr_Hisat2_index -1 $seq_data/T${i}_good_1.fq.gz -2 $seq_data/T${i}_good_2.fq.gz | $samtools sort -@ 8 -O bam -o $result/T${i}.sorted.bam
done
</code></pre>
<p>具体参数就不详讲了，具体见
<a href="http://ccb.jhu.edu/software/hisat2/manual.shtml" target="_blank" rel="noopener">文档</a>。</p>
<p>后续主要进行定量了，下次讲。</p>
    </div>

    






<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/rna-seq/">RNA-seq</a>
  
  <a class="badge badge-light" href="/tags/%E6%AF%94%E5%AF%B9/">比对</a>
  
</div>



<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=https://taoyan.netlify.app/post/2019-09-03.%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%9001/&amp;text=%e8%bd%ac%e5%bd%95%e7%bb%84%e6%95%b0%e6%8d%ae%e5%88%86%e6%9e%9001-%e6%af%94%e5%af%b9" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=https://taoyan.netlify.app/post/2019-09-03.%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%9001/&amp;t=%e8%bd%ac%e5%bd%95%e7%bb%84%e6%95%b0%e6%8d%ae%e5%88%86%e6%9e%9001-%e6%af%94%e5%af%b9" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=%e8%bd%ac%e5%bd%95%e7%bb%84%e6%95%b0%e6%8d%ae%e5%88%86%e6%9e%9001-%e6%af%94%e5%af%b9&amp;body=https://taoyan.netlify.app/post/2019-09-03.%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%9001/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=https://taoyan.netlify.app/post/2019-09-03.%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%9001/&amp;title=%e8%bd%ac%e5%bd%95%e7%bb%84%e6%95%b0%e6%8d%ae%e5%88%86%e6%9e%9001-%e6%af%94%e5%af%b9" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
  </ul>
</div>












  
  





  
    
    
    
      
    
    
    
    <div class="media author-card content-widget-hr">
      
        
        <img class="avatar mr-3 avatar-circle" src="/authors/alison/avatar_hufa245e75a5e5fa47274790393ba89624_26717_270x270_fill_q90_lanczos_center.jpg" alt="">
      

      <div class="media-body">
        <h5 class="card-title"><a href="https://taoyan.netlify.app/"></a></h5>
        <h6 class="card-subtitle">PhD Candidate of Crop Genetics and Breeding</h6>
        <p class="card-text">I am a PhD student of Crop Genetics and Breeding at the <em>Zhejiang University</em> Crop Science Lab. My research interests covers a range of issues:Population Genetics Evolution and Ecotype Divergence Analysis of Oilseed Rape, Genome-wide Association Study (GWAS) of Agronomic Traits.</p>
        <ul class="network-icon" aria-hidden="true">
  
    
    
    
      
    
    
    
    
    
    <li>
      <a href="mailto:tyan@zju.edu.cn" >
        <i class="fas fa-paper-plane"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
    <li>
      <a href="mailto:tyan@zju.edu.cn" >
        <i class="fas fa-envelope"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://twitter.com/TaoYANLogos" target="_blank" rel="noopener">
        <i class="fab fa-twitter"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://github.com/YTLogos" target="_blank" rel="noopener">
        <i class="fab fa-github"></i>
      </a>
    </li>
  
    
    
    
    
    
    
    
      
    
    <li>
      <a href="https://scholar.google.com/citations?hl=en&amp;user=8p3wQNwAAAAJ" target="_blank" rel="noopener">
        <i class="ai ai-google-scholar"></i>
      </a>
    </li>
  
</ul>

      </div>
    </div>
  







<section id="comments">
  
    
<div id="disqus_thread"></div>
<script>
  let disqus_config = function () {
    
    
    
  };
  (function() {
    if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
      document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
      return;
    }
    var d = document, s = d.createElement('script'); s.async = true;
    s.src = 'https://' + "logos-1" + '.disqus.com/embed.js';
    s.setAttribute('data-timestamp', +new Date());
    (d.head || d.body).appendChild(s);
  })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>


  
</section>










  </div>
</article>

      

    
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js" integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin="anonymous"></script>

      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js" integrity="sha256-eOgo0OtLL4cdq7RdwRUiGKLX9XsIJ7nGhWEKbohmVAQ=" crossorigin="anonymous"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/r.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/Shell.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/Python.min.js"></script>
        
      

      
      
    

    
    

    
    
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    
    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.37431be2d92d7fb0160054761ab79602.js"></script>

    






  
  
  <div class="container">
    <footer class="site-footer">
  
  <p class="powered-by">
    <a href="/license/">LICENSE: CC-BY-SA <br> <i class="fab fa-creative-commons fa-2x"></i><i class="fab fa-creative-commons-by fa-2x"></i><i class="fab fa-creative-commons-sa fa-2x"></i> </a><br>
  </p>
  

  <p class="powered-by">
    © Tao Yan, 2018-2020 &middot; 

    Made with <i class="far fa-heart" style="color:#E13D3D"></i>, <a href="https://github.com/YTLogos/TaoYan" target="_blank" rel="noopener"><i class="fas fa-code-branch" style="color:#E13D3D"></i></a>, the <a href="https://cran.r-project.org/" target="_blank" rel="noopener"><i class="fab fa-r-project"></i> </a><a href="https://github.com/rstudio/blogdown" target="_blank" rel="noopener">blogdown</a> package, and the 
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" id="back_to_top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

<div style="text-align: center;"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/6.js?i=5cz8zk3uynm&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=0" async="async"></script></div>

  </div>
  

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

</body>
</html>
