<!DOCTYPE html><html lang="en-us" >

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 4.8.0">

  

  
  
  
  
  
    
    
    
  
  

  

  
  
  
    
  
  <meta name="description" content="Tao Yan&#39;s Plot Toolkit">

  
  <link rel="alternate" hreflang="en-us" href="https://taoyan.netlify.app/project/ttplot/">

  


  
  
  
  <meta name="theme-color" content="#06805E">
  

  
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.0-1/css/all.min.css" integrity="sha256-4w9DunooKSr3MFXHXWyFER38WmPdm361bQS/2KUWZbU=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css" integrity="sha256-Vzbj7sDDS/woiFS3uNKo8eIuni59rjyNGtXfstRzStA=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/tomorrow.min.css" crossorigin="anonymous" title="hl-light">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/tomorrow.min.css" crossorigin="anonymous" title="hl-dark" disabled>
      
    

    

    

    
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js" integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin="anonymous" async></script>
      
    
      

      
      

      
    
      

      
      

      
    
      

      
      
        
      

      
    
      

      
      

      
    

  

  
  
  
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lora:400,400i%7CKarla:400,400i,700,700i%7CRoboto+Mono%7CKaushan+Script&display=swap">
  

  
  
  
  
  <link rel="stylesheet" href="/css/academic.css">

  





<script async src="https://www.googletagmanager.com/gtag/js?id=UA-99209585-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];

  function gtag() {
      dataLayer.push(arguments);
  }

  function trackOutboundLink(url, target) {
    gtag('event', 'click', {
         'event_category': 'outbound',
         'event_label': url,
         'transport_type': 'beacon',
         'event_callback': function () {
           if (target !== '_blank') {
             document.location = url;
           }
         }
    });
    console.debug("Outbound link clicked: " + url);
  }

  function onClickCallback(event) {
    if ((event.target.tagName !== 'A') || (event.target.host === window.location.host)) {
      return;
    }
    trackOutboundLink(event.target, event.target.getAttribute('target'));  
  }

  gtag('js', new Date());
  gtag('config', 'UA-99209585-1', {});

  
  document.addEventListener('click', onClickCallback, false);
</script>


  


  
  

  

  <link rel="manifest" href="/index.webmanifest">
  <link rel="icon" type="image/png" href="/images/icon_hubd7880285348c4660acc9402140a2483_242472_32x32_fill_lanczos_center_2.png">
  <link rel="apple-touch-icon" type="image/png" href="/images/icon_hubd7880285348c4660acc9402140a2483_242472_192x192_fill_lanczos_center_2.png">

  <link rel="canonical" href="https://taoyan.netlify.app/project/ttplot/">

  
  
  
  
  
  
  
    
  
  
  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@TaoYANLogos">
  <meta property="twitter:creator" content="@TaoYANLogos">
  
  <meta property="og:site_name" content="Tao Yan">
  <meta property="og:url" content="https://taoyan.netlify.app/project/ttplot/">
  <meta property="og:title" content="ttplot:Tao Yan&#39;s Plot Toolkit | Tao Yan">
  <meta property="og:description" content="Tao Yan&#39;s Plot Toolkit"><meta property="og:image" content="https://taoyan.netlify.app/project/ttplot/featured.png">
  <meta property="twitter:image" content="https://taoyan.netlify.app/project/ttplot/featured.png"><meta property="og:locale" content="en-us">
  
    
      <meta property="article:published_time" content="2020-09-14T00:00:00&#43;00:00">
    
    <meta property="article:modified_time" content="2020-09-14T00:00:00&#43;00:00">
  

  


    









<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "Article",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "https://taoyan.netlify.app/project/ttplot/"
  },
  "headline": "ttplot:Tao Yan's Plot Toolkit",
  
  "image": [
    "https://taoyan.netlify.app/project/ttplot/featured.png"
  ],
  
  "datePublished": "2020-09-14T00:00:00Z",
  "dateModified": "2020-09-14T00:00:00Z",
  
  "author": {
    "@type": "Person",
    "name": "Tao Yan"
  },
  
  "publisher": {
    "@type": "Organization",
    "name": "Tao Yan",
    "logo": {
      "@type": "ImageObject",
      "url": "https://taoyan.netlify.app/images/icon_hubd7880285348c4660acc9402140a2483_242472_192x192_fill_lanczos_center_2.png"
    }
  },
  "description": "Tao Yan's Plot Toolkit"
}
</script>

  

  


  


  





  <title>ttplot:Tao Yan&#39;s Plot Toolkit | Tao Yan</title>

</head>

<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" >

  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


  
<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">Tao Yan</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav ml-auto">
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/post"><span>Blog</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/publication"><span>Publications</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/projects"><span>Projects</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/CV.html"><span>CV</span></a>
        </li>

        
        

      

        

        
        <li class="nav-item">
          <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        

        

      </ul>

    </div>
  </div>
</nav>


  <article class="article article-project">

  




















  
  
    
  


<div class="article-container pt-3">
  <h1>ttplot:Tao Yan&#39;s Plot Toolkit</h1>

  

  



<meta content="2020-09-14 00:00:00 &#43;0000 UTC" itemprop="datePublished">
<meta content="2020-09-14 00:00:00 &#43;0000 UTC" itemprop="dateModified">

<div class="article-metadata">

  
  
  
  
  <div>
    

  
  <span>Tao Yan</span>
  </div>
  
  

  
  <span class="article-date">
    
    
      
    
    <time>Sep 14, 2020</time>
  </span>
  

  

  

  
  

  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fas fa-folder"></i>
    <a href="/categories/software/">software</a></span>
  

  
    
<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=https://taoyan.netlify.app/project/ttplot/&amp;text=ttplot:Tao%20Yan&amp;#39;s%20Plot%20Toolkit" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=https://taoyan.netlify.app/project/ttplot/&amp;t=ttplot:Tao%20Yan&amp;#39;s%20Plot%20Toolkit" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=ttplot:Tao%20Yan&amp;#39;s%20Plot%20Toolkit&amp;body=https://taoyan.netlify.app/project/ttplot/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=https://taoyan.netlify.app/project/ttplot/&amp;title=ttplot:Tao%20Yan&amp;#39;s%20Plot%20Toolkit" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
  </ul>
</div>


  

</div>
  













<div class="btn-links mb-3">
  
  








  






  
  
    
  
<a class="btn btn-outline-primary my-1 mr-1" href="https://github.com/YTLogos/ttplot" target="_blank" rel="noopener">
  Code
</a>














</div>


</div>


<div class="article-header article-container featured-image-wrapper mt-4 mb-4" style="max-width: 720px; max-height: 557px;">
  <div style="position: relative">
    <img src="/project/ttplot/featured_hud7ba50c55c24856fef46e3e1460229ef_481996_720x0_resize_lanczos_2.png" alt="" class="featured-image">
    <span class="article-header-caption">The home page</span>
  </div>
</div>



  <div class="article-container">

    <div class="article-style">
      <h2 id="overview">Overview</h2>
<p>This package is just developed to finish my project. So most of scripts are <strong>NOT stable</strong> enough. By wrapping some functions from other useful packages, it&rsquo;s easy for us to finish our jobs.</p>
<h2 id="installation">Installation</h2>
<p>This package dependents on some other packages:
<a href="https://github.com/knausb/vcfR" target="_blank" rel="noopener"><strong>vcfR</strong></a>,
<a href="https://github.com/tidyverse/dplyr" target="_blank" rel="noopener"><strong>dplyr</strong></a>,
<a href="https://cran.r-project.org/web/packages/genetics/index.html" target="_blank" rel="noopener"><strong>genetics</strong></a>,
<a href="https://github.com/SFUStatgen/LDheatmap" target="_blank" rel="noopener"><strong>LDheatmap</strong></a>, &hellip;</p>
<p>The dependent packages will be installed at the time you install the package 
<a href="https://github.com/YTLogos/ttplot" target="_blank" rel="noopener"><strong>ttplot</strong></a></p>
<pre><code>#install the ttplot package from Github
if(!require(devtools)) {
  install.packages(&quot;devtools&quot;); 
  require(devtools)
} 
devtools::install_github(&quot;YTLogos/ttplot&quot;)
</code></pre>
<h2 id="usage">Usage</h2>
<p>Until now there are just several functions</p>
<h3 id="draw-the-ldheatmap-from-the-vcfhttpsenwikipediaorgwikivariant_call_format-format-file-plinkhttpswwwcog-genomicsorgplink2-format">Draw the LDheatmap from the 
<a href="https://en.wikipedia.org/wiki/Variant_Call_Format" target="_blank" rel="noopener"><strong>VCF</strong></a> format file (
<a href="https://www.cog-genomics.org/plink2/" target="_blank" rel="noopener"><strong>plink</strong></a> format)</h3>
<p><strong>VCF</strong> (Variant Call Format) is a text file format. It contains meta-information lines, a header line, and then data lines each containing information about a position in the genome. There is an example how to draw LDheatmap from data in VCF file(<strong>plink</strong> format). The <strong>VCF</strong> file looks like:</p>
<pre><code>#This is a test vcf file (test.vcf) 

##fileformat=VCFv4.3
##fileDate=20090805
##source=myImputationProgramV3.1
##reference=file:///seq/references/1000GenomesPilot-NCBI36.fasta
##contig=&lt;ID=20,length=62435964,assembly=B36,md5=f126cdf8a6e0c7f379d618ff66beb2da,species=&quot;Homo sapiens&quot;,taxonomy=x&gt;
##phasing=partial
##INFO=&lt;ID=NS,Number=1,Type=Integer,Description=&quot;Number of Samples With Data&quot;&gt;
##INFO=&lt;ID=DP,Number=1,Type=Integer,Description=&quot;Total Depth&quot;&gt;
##INFO=&lt;ID=AF,Number=A,Type=Float,Description=&quot;Allele Frequency&quot;&gt;
##INFO=&lt;ID=AA,Number=1,Type=String,Description=&quot;Ancestral Allele&quot;&gt;
##INFO=&lt;ID=DB,Number=0,Type=Flag,Description=&quot;dbSNP membership, build 129&quot;&gt;
##INFO=&lt;ID=H2,Number=0,Type=Flag,Description=&quot;HapMap2 membership&quot;&gt;
##FILTER=&lt;ID=q10,Description=&quot;Quality below 10&quot;&gt;
##FILTER=&lt;ID=s50,Description=&quot;Less than 50% of samples have data&quot;&gt;
##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;
##FORMAT=&lt;ID=GQ,Number=1,Type=Integer,Description=&quot;Genotype Quality&quot;&gt;
##FORMAT=&lt;ID=DP,Number=1,Type=Integer,Description=&quot;Read Depth&quot;&gt;
##FORMAT=&lt;ID=HQ,Number=2,Type=Integer,Description=&quot;Haplotype Quality&quot;&gt;
#CHROM POS      ID         REF   ALT    QUAL  FILTER   INFO                             FORMAT       NA00001         NA00002          NA00003
20     14370    rs6054257  G     A      29    PASS    NS=3;DP=14;AF=0.5;DB;H2           GT:GQ:DP:HQ  0|0:48:1:51,51  1|0:48:8:51,51   1/1:43:5:.,.
20     17330    .          T     A      3     q10     NS=3;DP=11;AF=0.017               GT:GQ:DP:HQ  0|0:49:3:58,50  0|1:3:5:65,3     0/0:41:3
20     1110696  rs6040355  A     G,T    67    PASS    NS=2;DP=10;AF=0.333,0.667;AA=T;DB GT:GQ:DP:HQ  1|2:21:6:23,27  2|1:2:0:18,2     2/2:35:4
20     1230237  .          T     .      47    PASS    NS=3;DP=13;AA=T                   GT:GQ:DP:HQ  0|0:54:7:56,60  0|0:48:4:51,51   0/0:61:2
20     1234567  microsat1  GTC   G,GTCT 50    PASS    NS=3;DP=9;AA=G                    GT:GQ:DP     0/1:35:4        0/2:17:2         1/1:40:3
</code></pre>
<p>This kind of <strong>VCF</strong> is very large, so first we can use <strong>plink</strong> to recode the <strong>VCF</strong> file</p>
<pre><code>$ plink --vcf test.vcf --recode vcf-iid --out Test -allow-extra-chr
</code></pre>
<p>So the final <strong>VCF</strong> file we will use looks like：</p>
<pre><code>##fileformat=VCFv4.2
##fileDate=20180905
##source=PLINKv1.90
##contig=&lt;ID=chrC07,length=31087537&gt;
##INFO=&lt;ID=PR,Number=0,Type=Flag,Description=&quot;Provisional reference allele, may not be based on real reference genome&quot;&gt;
##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	R4157	R4158
chrC07	31076164	.	T	G	.	.	PR	GT	0/0	0/1
chrC07	31076273	.	G	A	.	.	PR	GT	0/0	0/1
chrC07	31076306	.	G	T	.	.	PR	GT	0/0	0/0
</code></pre>
<p>In the <code>extdata</code> directory there is one test file: <code>test.vcf</code>. We can test the function:</p>
<pre><code>library(ttplot)
test &lt;- system.file(&quot;extdata&quot;, &quot;test.vcf&quot;, package = &quot;ttplot&quot;)
ttplot::MyLDheatMap(vcffile = test, title = &quot;My gene region&quot;)
</code></pre>
<p><img src="https://raw.githubusercontent.com/YTLogos/Pic_blog/master/LDheatmap.png" alt="LDheatmap"></p>
<h3 id="more">More</h3>
<h4 id="usage-1"><strong>Usage:</strong></h4>
<pre><code>MyLDheatMap(vcffile, file.output = TRUE, file = &quot;png&quot;,
  output = &quot;region&quot;, title = &quot;region:&quot;, verbose = TRUE, dpi = 300)
</code></pre>
<h4 id="arguments">Arguments</h4>
<ul>
<li><strong>vcffile:</strong> The plink format vcf file. More detail can see <code>View(test_vcf)</code>.</li>
<li><strong>file.output:</strong> a logical, if <code>file.output=TRUE</code>, the result will be saved. if <code>file.output=FALSE</code>, the result will be printed. The default is <code>TRUE</code>.</li>
<li><strong>file:</strong> a character, users can choose the different output formats of plot, so far, <strong>&ldquo;jpeg&rdquo;</strong>, <strong>&ldquo;pdf&rdquo;</strong>, <strong>&ldquo;png&rdquo;</strong>, <strong>&ldquo;tiff&rdquo;</strong> can be selected by users. The default is <strong>&ldquo;png&rdquo;</strong>.</li>
<li><strong>title:</strong>	a character, the title of the LDheatmap will be &ldquo;The LDheatmap of title&rdquo;. the default is &ldquo;region:&rdquo;. I suggest users use your own title.</li>
<li><strong>verbose:</strong>	whether to print the reminder.</li>
<li><strong>dpi:</strong>	a number, the picture element for <strong>.jpeg</strong>, <strong>.png</strong> and <strong>.tiff</strong> files. The default is <code>300</code>.</li>
</ul>
<h2 id="ggmanhattan-make-manhattan-plot-based-on-ggplot2">ggmanhattan(): make manhattan plot based on <strong>ggplot2</strong></h2>
<p>Here, we will use the data <code>gwasResults</code> from R package 
<a href="https://github.com/stephenturner/qqman" target="_blank" rel="noopener"><strong>qqman</strong></a>. This function is provided to make manhattan plot with full ggplot customizability. So next we can customize the manhattan plot with kinds of functions of <strong>ggplot2</strong> and add additional layers. The data need be reshaped as following:</p>
<pre><code>&gt; head(gwasResults)
  SNP CHR BP         P
1 rs1   1  1 0.9148060
2 rs2   1  2 0.9370754
3 rs3   1  3 0.2861395
4 rs4   1  4 0.8304476
5 rs5   1  5 0.6417455
6 rs6   1  6 0.5190959
</code></pre>
<p>Then use the function <code>ggmanhattan</code> from 
<a href="https://github.com/YTLogos/ttplot" target="_blank" rel="noopener"><strong>ttplot</strong></a> to make a manhattan plot:</p>
<pre><code>ttplot::ggmanhattan(gwasResults)
</code></pre>
<p><img src="https://raw.githubusercontent.com/YTLogos/Pic_blog/master/The%20Manhattan%20Plot%20of%20gwas.png" alt=""></p>
<h3 id="more-1">More</h3>
<pre><code>ggmanhattan(gwasres, snp = NA, bp = NA, chrom = NA, pvalue = NA,
  index = NA, file.output = FALSE, file = &quot;png&quot;, output = &quot;Trait&quot;,
  dpi = 300, vlinetype = &quot;solid&quot;, vlinesize = 0.75,
  title = &quot;Manhattan Plot&quot;, color = c(&quot;#FF8C00&quot;, &quot;#556B2F&quot;),
  pointsize = 1.25, verbose = TRUE, ...)
</code></pre>
<h4 id="arguments-1">Arguments</h4>
<ul>
<li><strong>gwasres</strong>	a data frame of gwas results.</li>
<li><strong>snp</strong>	Name of the column containing <code>SNP</code> identifiers; default is &lsquo;snp&rsquo;.</li>
<li><strong>bp</strong>	Name of the column containing the <code>SNP</code> positions; default is <code>'bp'</code>.</li>
<li><strong>chrom</strong>	Name of the column containing the chromosome identifers; default is <code>'chrom'</code>.</li>
<li><strong>pvalue</strong>	Name of the column containing the p values; default is <code>'pvalue'</code>.</li>
<li><strong>file.output</strong>	a logical, if file.output=<code>TRUE</code>, the result will be saved. if file.output=<code>FALSE</code>, the result will be printed. The default is <code>TRUE</code>.</li>
<li><strong>file</strong>	a character, users can choose the different output formats of plot, so far, <code>&quot;jpeg&quot;</code>, <code>&quot;pdf&quot;</code>, <code>&quot;png&quot;</code>, <code>&quot;tiff&quot;</code> can be selected by users. The default is <code>&quot;png&quot;</code>.</li>
<li><strong>dpi</strong>	a number, the picture element for .jpeg, .png and .tiff files. The default is 300.</li>
<li><strong>vlinetype</strong>	the type of vline (<code>geom_vline()</code>). The default is <code>&quot;solid&quot;</code>.</li>
<li><strong>vlinesize</strong>	the size of the vline. The default is 0.75.</li>
<li><strong>title</strong>	the title of manhattan plot. The default is <code>&quot;Manhattan Plot&quot;</code>.</li>
<li><strong>color</strong>	the colors of alternate chromosome. The default is c(<code>&quot;#FF8C00&quot;</code>, <code>&quot;#556B2F&quot;</code>).</li>
<li><strong>pointsize</strong>	the size of point. The default is 1.25.</li>
</ul>
<h2 id="get-genes-bsaed-on-the-locations-of-significant-snps">Get genes bsaed on the locations of significant SNPs</h2>
<p>This function need two input files: gff, snp.</p>
<pre><code>library(ttplot)
gff &lt;- read.table(system.file(&quot;extdata&quot;, &quot;test.gff&quot;, package = &quot;ttplot&quot;), header= TRUE)
head(gff)
</code></pre>
<pre><code>chr  start    end          gene
1 436789 437474 chrA01g000077
1 439907 442623 chrA01g000078
1 448692 449999 chrA01g000079
1 454920 456931 chrA01g000080
1 457568 460619 chrA01g000081
1 461129 462693 chrA01g000082
</code></pre>
<pre><code>snp &lt;- read.table(system.file(&quot;extdata&quot;, &quot;test_sig_snp.txt&quot;, package = &quot;ttplot&quot;), header= TRUE)
head(snp)
</code></pre>
<pre><code>SNP CHR       BP        P
1_500000   1   500000 5.53e-11
1_650000   1   650000 7.04e-09
2_1880000   2  1880000 3.84e-09
3_30500000   3 30500000 7.57e-08
</code></pre>
<p>Then we can get the candidate genes in regions based on significant SNPs of GWAS results.</p>
<pre><code>ttplot::get_gene_from_snp(gff = gff, sig.snp = snp, distance = 20000, file.save = FALSE)
The distance you choose is 20000bp!
You have 4 significant SNPs and 130 genes!
Now we will extract the genes in the significant regions! This will need some time, please wait for severals minutes! ...
                                 
The result is : 
-------------------------------------------------------------
# A tibble: 9 x 5
chrom geneid        gene_start gene_end snp_location
&lt;dbl&gt; &lt;chr&gt;              &lt;dbl&gt;    &lt;dbl&gt;        &lt;dbl&gt;
1 chrA01g000088     479923   485997       500000
1 chrA01g000089     495531   497917       500000
1 chrA01g000090     498488   499689       500000
1 chrA01g000091     499749   501301       500000
1 chrA01g000092     503622   507079       500000
1 chrA01g000093     507659   510321       500000
1 chrA01g000094     515103   516863       500000
1 chrA01g000095     517335   518776       500000
1 chrA01g000096     519020   520668       500000
</code></pre>
<h3 id="more-2">More</h3>
<pre><code>Usage
get_gene_from_snp(gff, sig.snp, distance = 50000, file.save = TRUE,
  file.type = &quot;csv&quot;, gff.chrom = NA, snp.chrom = NA, geneid = NA,
  pvalue = NA, gene_start = NA, gene_end = NA, snp_location = NA,
  verbose = TRUE, ...)
</code></pre>
<h4 id="arguments-2">Arguments</h4>
<ul>
<li>gff: a data frame of all the gene (transcript), must have column names.</li>
<li>sig.snp: a data frame of significant SNPs.</li>
<li>distance: numeric (bp), it is to define the region. The default is 50000, you need to choose it based on the LD distance in your study.</li>
<li>file.save:	a logical, if file.output=TRUE, the result will be saved. if file.output=FALSE, the result will be printed. The default is TRUE.</li>
<li>file.type: a character, users can choose the different output formats, so far, &ldquo;csv&rdquo;, &ldquo;txt&rdquo;, &ldquo;xlsx&rdquo; can be selected by users. The default is &ldquo;csv&rdquo;.</li>
<li>gff.chrom: Name of the column containing the chromosome identifers in the gff file; default is NA.</li>
<li>snp.chrom: Name of the column containing the chromosome identifers in the snp.sig file; default is NA.</li>
<li>geneid: Name of the column containing the geneid in gff file; default is NA.</li>
<li>pvalue: Name of the column containing the p values in snp.sig file; default is NA.</li>
<li>snp_location: Name of the column containing the snp position in snp.sig file; default is NA.</li>
</ul>

    </div>

    






<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/software/">software</a>
  
  <a class="badge badge-light" href="/tags/r/">R</a>
  
</div>



<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=https://taoyan.netlify.app/project/ttplot/&amp;text=ttplot:Tao%20Yan&amp;#39;s%20Plot%20Toolkit" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=https://taoyan.netlify.app/project/ttplot/&amp;t=ttplot:Tao%20Yan&amp;#39;s%20Plot%20Toolkit" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=ttplot:Tao%20Yan&amp;#39;s%20Plot%20Toolkit&amp;body=https://taoyan.netlify.app/project/ttplot/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=https://taoyan.netlify.app/project/ttplot/&amp;title=ttplot:Tao%20Yan&amp;#39;s%20Plot%20Toolkit" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
  </ul>
</div>












  
    
    





  


  





<section id="comments">
  
    
<div id="disqus_thread"></div>
<script>
  let disqus_config = function () {
    
    
    
  };
  (function() {
    if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
      document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
      return;
    }
    var d = document, s = d.createElement('script'); s.async = true;
    s.src = 'https://' + "logos-1" + '.disqus.com/embed.js';
    s.setAttribute('data-timestamp', +new Date());
    (d.head || d.body).appendChild(s);
  })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>


  
</section>










    <div class="project-related-pages content-widget-hr">
      
      

      
      
      

      
      
      

      
      
      
    </div>
  </div>
</article>

      

    
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js" integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin="anonymous"></script>

      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js" integrity="sha256-eOgo0OtLL4cdq7RdwRUiGKLX9XsIJ7nGhWEKbohmVAQ=" crossorigin="anonymous"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/r.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/Shell.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/Python.min.js"></script>
        
      

      
      
    

    
    

    
    
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    
    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.37431be2d92d7fb0160054761ab79602.js"></script>

    






  
  
  <div class="container">
    <footer class="site-footer">
  
  <p class="powered-by">
    <a href="/license/">LICENSE: CC-BY-SA <br> <i class="fab fa-creative-commons fa-2x"></i><i class="fab fa-creative-commons-by fa-2x"></i><i class="fab fa-creative-commons-sa fa-2x"></i> </a><br>
  </p>
  

  <p class="powered-by">
    © Tao Yan, 2018-2020 &middot; 

    Made with <i class="far fa-heart" style="color:#E13D3D"></i>, <a href="https://github.com/YTLogos/TaoYan" target="_blank" rel="noopener"><i class="fas fa-code-branch" style="color:#E13D3D"></i></a>, the <a href="https://cran.r-project.org/" target="_blank" rel="noopener"><i class="fab fa-r-project"></i> </a><a href="https://github.com/rstudio/blogdown" target="_blank" rel="noopener">blogdown</a> package, and the 
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" id="back_to_top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

<div style="text-align: center;"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/6.js?i=5cz8zk3uynm&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=0" async="async"></script></div>

  </div>
  

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

</body>
</html>
