<!DOCTYPE html>
<html lang="en-us">

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 4.4.0">

  

  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Tao Yan">

  
  
  
    
  
  <meta name="description" content="

文件准备

利用EMMAX进行GWAS分析需要以下文件">

  
  <link rel="alternate" hreflang="en-us" href="https://taoyan.netlify.com/post/2019-02-27.%E5%88%A9%E7%94%A8emmax%E8%BF%9B%E8%A1%8Cgwas%E5%88%86%E6%9E%90/">

  


  

  
  
  
  <meta name="theme-color" content="#E13D3D">
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.0/css/all.css" integrity="sha384-aOkxzJ5uQz7WBObEZcHvV5JvRW3TUc2rNPA7pe3AwnsUohiw1Vj2Rgx2KSOkF5+h" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/paraiso-light.min.css" crossorigin="anonymous" title="hl-light">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/paraiso-light.min.css" crossorigin="anonymous" title="hl-dark" disabled>
      
    

    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.2.0/leaflet.css" integrity="sha512-M2wvCLH6DSRazYeZRIm1JnYyh22purTM+FDB5CsyxtQJYeKq83arPe5wgbNmcFXGqiSH2XR8dT/fJISVA1r/zQ==" crossorigin="anonymous">
    

    

  

  
  
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Arapey:400,400i|Karla:400,700|Roboto+Mono|Mrs+Saint+Delafield|Kaushan+Script|Playball&display=swap">
  

  
  
  
  <link rel="stylesheet" href="/css/academic.min.271ffa4140fad041464911a0b7fff4cf.css">

  
    
    
    
    
      
    
    
    
    <link rel="stylesheet" href="/css/academic.948c91384be7a6e3e086b01b867a93a8.css">
  

  
  
  

  

  <link rel="manifest" href="/index.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon-32.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="https://taoyan.netlify.com/post/2019-02-27.%E5%88%A9%E7%94%A8emmax%E8%BF%9B%E8%A1%8Cgwas%E5%88%86%E6%9E%90/">

  
  
  
  
    
  
  
  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@TaoYANLogos">
  <meta property="twitter:creator" content="@TaoYANLogos">
  
  <meta property="og:site_name" content="Tao Yan">
  <meta property="og:url" content="https://taoyan.netlify.com/post/2019-02-27.%E5%88%A9%E7%94%A8emmax%E8%BF%9B%E8%A1%8Cgwas%E5%88%86%E6%9E%90/">
  <meta property="og:title" content="利用EMMAX进行GWAS分析 | Tao Yan">
  <meta property="og:description" content="

文件准备

利用EMMAX进行GWAS分析需要以下文件"><meta property="og:image" content="https://taoyan.netlify.com/img/avatar.jpg">
  <meta property="twitter:image" content="https://taoyan.netlify.com/img/avatar.jpg"><meta property="og:locale" content="en-us">
  
    
      <meta property="article:published_time" content="2019-02-26T18:19:05&#43;00:00">
    
    <meta property="article:modified_time" content="2019-02-26T18:19:05&#43;00:00">
  

  

<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.0.3/cookieconsent.min.css">
<script src="https://cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.0.3/cookieconsent.min.js"></script>
<script>
  window.addEventListener("load", function(){
    window.cookieconsent.initialise({
      "palette": {
        "popup": {
          "background": "#E13D3D",
          "text": "#fbf9f4"
        },
        "button": {
          "background": "#fbf9f4",
          "text": "#E13D3D"
        }
      },
      "theme": "classic",
      "content": {
        "message": "This website uses cookies to ensure you get the best experience on our website.",
        "dismiss": "Got it!",
        "link": "Learn more",
        "href": "https://cookies.insites.com"
      }
    })});
</script>



  





  <title>利用EMMAX进行GWAS分析 | Tao Yan</title>

</head>

<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" >

  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


  
<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0 compensate-for-scrollbar" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">Tao Yan</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav ml-auto">
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#posts"><span>Posts</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#projects"><span>Projects</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#featured"><span>Publications</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/courses/"><span>Courses</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#contact"><span>Contact</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/files/CV.pdf"><span>CV</span></a>
        </li>

        
        

      

        

        
        <li class="nav-item">
          <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        

        
        <li class="nav-item">
          <a class="nav-link js-dark-toggle" href="#"><i class="fas fa-moon" aria-hidden="true"></i></a>
        </li>
        

      </ul>

    </div>
  </div>
</nav>


  <article class="article" itemscope itemtype="http://schema.org/Article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1 itemprop="name">利用EMMAX进行GWAS分析</h1>

  

  
    



<meta content="2019-02-26 18:19:05 &#43;0000 UTC" itemprop="datePublished">
<meta content="2019-02-26 18:19:05 &#43;0000 UTC" itemprop="dateModified">

<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
    
    <time>Feb 26, 2019</time>
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    4 min read
  </span>
  

  
  

  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fas fa-folder"></i>
    <a href="/categories/bioinformatics/">Bioinformatics</a></span>
  

  
    
<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=https://taoyan.netlify.com/post/2019-02-27.%E5%88%A9%E7%94%A8emmax%E8%BF%9B%E8%A1%8Cgwas%E5%88%86%E6%9E%90/&amp;text=%e5%88%a9%e7%94%a8EMMAX%e8%bf%9b%e8%a1%8cGWAS%e5%88%86%e6%9e%90" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=https://taoyan.netlify.com/post/2019-02-27.%E5%88%A9%E7%94%A8emmax%E8%BF%9B%E8%A1%8Cgwas%E5%88%86%E6%9E%90/&amp;t=%e5%88%a9%e7%94%a8EMMAX%e8%bf%9b%e8%a1%8cGWAS%e5%88%86%e6%9e%90" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook-f"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=%e5%88%a9%e7%94%a8EMMAX%e8%bf%9b%e8%a1%8cGWAS%e5%88%86%e6%9e%90&amp;body=https://taoyan.netlify.com/post/2019-02-27.%E5%88%A9%E7%94%A8emmax%E8%BF%9B%E8%A1%8Cgwas%E5%88%86%E6%9E%90/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=https://taoyan.netlify.com/post/2019-02-27.%E5%88%A9%E7%94%A8emmax%E8%BF%9B%E8%A1%8Cgwas%E5%88%86%E6%9E%90/&amp;title=%e5%88%a9%e7%94%a8EMMAX%e8%bf%9b%e8%a1%8cGWAS%e5%88%86%e6%9e%90" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://web.whatsapp.com/send?text=%e5%88%a9%e7%94%a8EMMAX%e8%bf%9b%e8%a1%8cGWAS%e5%88%86%e6%9e%90%20https://taoyan.netlify.com/post/2019-02-27.%E5%88%A9%E7%94%A8emmax%E8%BF%9B%E8%A1%8Cgwas%E5%88%86%E6%9E%90/" target="_blank" rel="noopener" class="share-btn-whatsapp">
          <i class="fab fa-whatsapp"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://service.weibo.com/share/share.php?url=https://taoyan.netlify.com/post/2019-02-27.%E5%88%A9%E7%94%A8emmax%E8%BF%9B%E8%A1%8Cgwas%E5%88%86%E6%9E%90/&amp;title=%e5%88%a9%e7%94%a8EMMAX%e8%bf%9b%e8%a1%8cGWAS%e5%88%86%e6%9e%90" target="_blank" rel="noopener" class="share-btn-weibo">
          <i class="fab fa-weibo"></i>
        </a>
      </li>
    
  </ul>
</div>


  

</div>
    














  
</div>



  <div class="article-container">

    <div class="article-style" itemprop="articleBody">
      <p><img src="https://raw.githubusercontent.com/YTLogos/Pic_blog/master/manhanttan.jpg" alt="" /></p>

<h2 id="文件准备">文件准备</h2>

<p>利用<strong>EMMAX</strong>进行<strong>GWAS</strong>分析需要以下文件</p>

<h3 id="基因型文件">基因型文件</h3>

<p>基因型文件直接使用<strong>VCF</strong>进行过滤后得到，具体如下(我这里原始的<strong>VCF</strong>文件是<strong>test.vcf</strong>)，过滤前先将染色体名称全部换为数值型(不然后续<strong>GWAS</strong>分析会报错的)，同时添加<code>SNP ID</code>，这个比较简单就不讲了！
另外如果测序深度较低，可以基因型填补一下，按研究需要吧，这里就不进行了。</p>

<ul>
<li><p>按基因型等位频率(0.05)以及缺失率(0.1)进行过滤</p>

<pre><code>plink --vcf test.vcf --maf 0.05 --geno 0.1 --recode vcf-iid --out test.maf0.05.int0.9
</code></pre></li>
</ul>

<p>生成<strong>test.maf0.05.int0.9.vcf</strong>文件</p>

<ul>
<li><p>依据LD对标记进行筛选(这一步SNP必须有ID才行，GATK等软件生成的VCF文件是没有ID的)</p>

<pre><code>plink --vcf test.maf0.05.int0.9.vcf --indep-pairwise 50 10 0.2 --out test.maf0.05.int0.9
</code></pre></li>
</ul>

<p>产生<strong>test.maf0.05.int0.9.prune.in</strong>和<strong>test.maf0.05.int0.9.prune.out</strong>两个文件，<strong>in</strong>表示入选的标记，<strong>out</strong>表示被去除的标记，这两个文件都是只含有SNP ID，根据ID进行筛选</p>

<ul>
<li><p>提取筛选结果</p>

<pre><code>plink --vcf test.maf0.05.int0.9.vcf --make-bed --extract test.maf0.05.int0.9.prune.in --out test.maf0.05.int0.9.prune.in
</code></pre></li>

<li><p>将筛选后的数据转换为<strong>vcf</strong>格式</p>

<pre><code>plink --bfile test.maf0.05.int0.9.prune.in --recode vcf-iid --out test.maf0.05.int0.9.prune.in
</code></pre></li>
</ul>

<p>得到<strong>test.maf0.05.int0.9.prune.in.vcf</strong>基因型文件</p>

<ul>
<li><p>将筛选后的文件转换为<strong>admixture</strong>数据格式用于群体结构分析</p>

<pre><code>plink -bfile test.maf0.05.int0.9.prune.in --recode 12 --out test.maf0.05.int0.9.prune.in
</code></pre></li>
</ul>

<p>得到<strong>test.maf0.05.int0.9.prune.in.ped</strong>用于<strong>admixture</strong>分析</p>

<ul>
<li><p>将基因型文件转换为<strong>EMMAX</strong>格式</p>

<pre><code>plink --vcf test.maf0.05.int0.9.vcf --recode 12 transpose --out-missing-genotype 0 --out test.maf0.05.int0.9 --autosome-num 90
</code></pre></li>
</ul>

<p>得到<strong>tfam</strong>文件</p>

<ul>
<li><p>群体亲缘关系分析</p>

<pre><code>emmax-kin test.maf0.05.int0.9 -v -h -d 10 -o test.maf0.05.int0.9.BN.kinf
</code></pre></li>
</ul>

<h3 id="群体结构文件">群体结构文件</h3>

<p>首先<strong>admixture</strong>群体结构分析</p>

<pre><code>for i in {1..15}
do
admixture --cv test.maf0.05.int0.9.prune.in.ped ${i} &gt;&gt; log.txt
done
</code></pre>

<p>查看最佳分群</p>

<pre><code>$ grep CV log.txt
CV error (K=1): 0.43802
CV error (K=2): 0.41492
CV error (K=3): 0.40182
CV error (K=4): 0.39723
CV error (K=5): 0.39281
CV error (K=6): 0.38899
CV error (K=7): 0.38640
CV error (K=8): 0.38434
CV error (K=9): 0.38301
CV error (K=10): 0.38156
CV error (K=11): 0.38050
CV error (K=12): 0.38006
CV error (K=13): 0.37869
CV error (K=14): 0.37795
CV error (K=15): 0.37830
</code></pre>

<p>K=14时CV值最小，所以最佳分群为14
K=14时候的Q矩阵用于后续分析。<strong>admixture</strong>产生的Q矩阵格式如下(前6行)</p>

<pre><code>0.000010 0.864016 0.039019 0.000010 0.000010 0.000010 0.000010 0.000010 0.000010 0.096855 0.000010 0.000010 0.000010 0.000010
0.000010 0.135428 0.000010 0.039195 0.000010 0.192606 0.195391 0.062418 0.000010 0.124034 0.214463 0.000010 0.028383 0.008032
0.000010 0.613532 0.000010 0.043222 0.000010 0.000010 0.272258 0.000010 0.000010 0.000010 0.058748 0.000010 0.012151 0.000010
0.000010 0.419498 0.417594 0.000010 0.000010 0.000010 0.000010 0.000010 0.000010 0.000010 0.000010 0.090515 0.062396 0.009908
0.012456 0.045110 0.000010 0.000010 0.000010 0.021304 0.202956 0.049112 0.001099 0.455700 0.011129 0.093843 0.107251 0.000010
0.000010 0.071372 0.004548 0.000010 0.004779 0.000010 0.326421 0.009612 0.003101 0.211292 0.350671 0.000010 0.018153 0.000010
</code></pre>

<p>因为文件相对较小，手动添加表头信息，<strong>以及删除最后一列</strong>，保存<strong>14.Q</strong>用于后续分析,之后如下：</p>

<pre><code>&lt;Covariate&gt;
&lt;Trait&gt; Q1      Q2      Q3      Q4      Q5      Q6      Q7      Q8      Q9      Q10     Q11     Q12     Q13
R4155   0.00001 0.864016        0.039019        0.00001 0.00001 0.00001 0.00001 0.00001 0.00001 0.096855        0.00001 0.00001 0.00001
R4156   0.00001 0.135428        0.00001 0.039195        0.00001 0.192606        0.195391        0.062418        0.00001 0.124034        0.214463        0.00001 0.028383
R4157   0.00001 0.613532        0.00001 0.043222        0.00001 0.00001 0.272258        0.00001 0.00001 0.00001 0.058748        0.00001 0.012151
R4158   0.00001 0.419498        0.417594        0.00001 0.00001 0.00001 0.00001 0.00001 0.00001 0.00001 0.00001 0.090515        0.062396
</code></pre>

<ul>
<li>表型数据</li>
</ul>

<p><strong>EMMAX</strong>每次只能运行一个性状，格式与<strong>tfam</strong>格式类似，所以需要进行拆分转换，拆分前，需要将缺失值-999替换为NA。
最开始我的表型数据如下：</p>

<pre><code>&lt;Trait&gt; day
R4155   184
R4156   181
R4157   181
R4158   182
R4159   170
R4160   179
</code></pre>

<p>利用<strong>R</strong>进行转换</p>

<pre><code>tfam &lt;- read.table(&quot;test.maf0.05.int0.9.tfam&quot;, header = F, stringsAsFactors = F)
tr &lt;- read.table(&quot;test.trait.txt&quot;, header = T, check.names = F, stringsAsFactors = F)
tr &lt;- tr[match(tfam$V1, tr$`&lt;Trait&gt;`),] ###把基因型文件样本顺序和表型文件样本顺序调整为一致
tr[tr == -999] &lt;- NA
tre &lt;- cbind(tr[,1], tr)
write.table(tre, file = &quot;test_trait_emmax.txt&quot;, col.names = F, row.names = F, sep = &quot;\t&quot;, quote = F)
</code></pre>

<p>转换后结果如下:</p>

<pre><code>R4155   R4155   184
R4156   R4156   181
R4157   R4157   181
R4158   R4158   182
R4159   R4159   170
</code></pre>

<ul>
<li>协变量文件</li>
</ul>

<p>这里以群体结构作为协变量</p>

<pre><code>tfam &lt;- read.table(&quot;test.maf0.05.int0.9.tfam&quot;, header = F, stringsAsFactors = F)
admix &lt;- read.table(&quot;14.Q&quot;, header = T, check.names = F, stringsAsFactors = F, skip = 1)
admix &lt;- admix[match(tfam$V1, admix$`&lt;Trait&gt;`), ]
admix &lt;- cbind(admix[,1], admix[,1], rep(1, nrow(admix)), admix[,-1]) 
write.table(, file = &quot;test.emmax.cov.txt&quot;, col.names = F, row.names = F, sep = &quot;\t&quot;, quote = F)
</code></pre>

<p>生成的协变量文件格式如下：</p>

<pre><code>R4155   R4155   1       1e-05   0.864016        0.039019        1e-05   1e-05   1e-05   1e-05   1e-05   1e-05   0.096855        1e-05   1e-05   1e-05
R4156   R4156   1       1e-05   0.135428        1e-05   0.039195        1e-05   0.192606        0.195391        0.062418        1e-05   0.124034        0.214463        1e-05   0.028383
R4157   R4157   1       1e-05   0.613532        1e-05   0.043222        1e-05   1e-05   0.272258        1e-05   1e-05   1e-05   0.058748        1e-05   0.012151
R4158   R4158   1       1e-05   0.419498        0.417594        1e-05   1e-05   1e-05   1e-05   1e-05   1e-05   1e-05   1e-05   0.090515        0.062396
R4159   R4159   1       0.012456        0.04511 1e-05   1e-05   1e-05   0.021304        0.202956        0.049112        0.001099        0.4557  0.011129        0.093843 0.10725
</code></pre>

<h2 id="gwas-运行"><strong>GWAS</strong>运行</h2>

<p>所有准备文件都有之后，就可以进行<strong>GWAS</strong>分析了</p>

<pre><code>emmax -v -d 10 -t test.maf0.05.int0.9 -o test_emmax_cov -p test_trait_emmax.txt -k test.maf0.05.int0.9.BN.kinf -c test.cov.emmax.txt
</code></pre>

<p>最后生成的<strong>test_emmax_cov.ps</strong>就是我们需要的</p>

<h2 id="可视化">可视化</h2>

<p>这里用<strong>qqman</strong>进行manhattan以及<strong>QQ</strong>图的绘制</p>

<pre><code>#gwas_visualization.R
setwd(&quot;/database/pop-analysis-NY7/result/pop-structure/gwas&quot;)
if(!require(qqman)){
    install.packages(&quot;qqman&quot;)
    library(qqman)
}
if(!require(tidyverse)){
    install.packages(&quot;tidyverse&quot;)
    library(tidyverse)
}
gwas &lt;- read.table(&quot;test_emmax_cov.ps&quot;, header = FALSE, stringsAsFactors = FALSE)
colnames(gwas) &lt;- c(&quot;SNP&quot;,&quot;SE&quot;,&quot;P&quot;)
splite_chr &lt;- function(data,c){
  c(str_split(data[c], pattern = &quot;_&quot;)[[1]][1])
}
splite_pos &lt;- function(data,c){
  c(str_split(data[c], pattern = &quot;_&quot;)[[1]][2])
}
gwas$CHR &lt;- apply(gwas,1,splite_chr,c=&quot;SNP&quot;)
gwas$BP &lt;- apply(gwas,1,splite_pos,c=&quot;SNP&quot;)
gwas &lt;- gwas%&gt;%select(SNP,CHR,BP,P)
write.table(gwas,&quot;gwas_emmax_test_for_vis.ps&quot;, col.names = TRUE, row.names = FALSE, quote = FALSE)
pdf(&quot;test_emmax_manhattan.pdf&quot;,width = 12,height = 6)
manhattan(gwas)
dev.off()
pdf(&quot;test_emmax_qq.pdf&quot;, width = 8, height = 6)
qq(gwas$P)
dev.off()
png(&quot;test_emmax_manhattan.png&quot;,width = 960,height = 600,type = &quot;cairo&quot;)
manhattan(gwas)
dev.off()
png(&quot;test_emmax_qq.png&quot;, width = 800, height = 800,type = &quot;cairo&quot;)
qq(gwas$P)
dev.off()
</code></pre>

<p>将上述脚本保存为<code>gwas_visualization.R</code>，之后在<strong>linux</strong>中运行</p>

<pre><code>Rscript gwas_visualization.R
</code></pre>

<p>就可以得到我们需要的图</p>

<h2 id="后续分析">后续分析</h2>

<p>后续还有很多分析，比如，显著位点的提取，关联基因的提取，关联基因的注释等等。。。。。。。</p>
    </div>

    
<p class="edit-page">
  <a href="https://github.com/YTLogos/TaoYan/edit/master/content/post%5c2019-02-27.%e5%88%a9%e7%94%a8EMMAX%e8%bf%9b%e8%a1%8cGWAS%e5%88%86%e6%9e%90.md">
    <i class="fas fa-pen pr-2"></i>Edit this page
  </a>
</p>



    

<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/gwas/">GWAS</a>
  
</div>



    
      








  





  
  
  
    
  
  
  <div class="media author-card" itemscope itemtype="http://schema.org/Person">
    
      
      <img class="portrait mr-3" src="/authors/admin/avatar_hufa245e75a5e5fa47274790393ba89624_26717_250x250_fill_q90_lanczos_center.jpg" itemprop="image" alt="Avatar">
    

    <div class="media-body">
      <h5 class="card-title" itemprop="name"><a href="https://taoyan.netlify.com">Tao Yan</a></h5>
      <h6 class="card-subtitle">PhD Candidate of Crop Science</h6>
      <p class="card-text" itemprop="description">My research interests covers a range of issues:Population Genetics Evolution and Ecotype Divergence Analysis of oilseed Rape, Genome-wide Association Study (GWAS) of Agronomic Traits.</p>
      <ul class="network-icon" aria-hidden="true">
        
          
          
          
            
          
          
          
          
          
          <li>
            <a itemprop="sameAs" href="mailto:tyan@zju.edu.cn" >
              <i class="fas fa-paper-plane"></i>
            </a>
          </li>
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a itemprop="sameAs" href="/#contact" >
              <i class="fas fa-envelope"></i>
            </a>
          </li>
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a itemprop="sameAs" href="https://twitter.com/TaoYANLogos" target="_blank" rel="noopener">
              <i class="fab fa-twitter"></i>
            </a>
          </li>
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a itemprop="sameAs" href="https://github.com/YTLogos" target="_blank" rel="noopener">
              <i class="fab fa-github"></i>
            </a>
          </li>
        
      </ul>
    </div>
  </div>



      
      
      <div class="article-widget">
        <div class="hr-light"></div>
        <h3>Related</h3>
        <ul>
          
          <li><a href="/post/2018-01-11.%E4%B8%80%E6%AD%A5%E6%AD%A5%E6%95%99%E4%BD%A0%E5%A6%82%E4%BD%95%E7%BB%98%E5%88%B6gwas%E4%B8%AD%E7%9A%84%E6%9B%BC%E5%93%88%E9%A1%BF%E5%9B%BE%E4%BB%A5%E5%8F%8Aqq%E5%9B%BE/">一步步教你如何绘制GWAS中的曼哈顿图以及QQ图</a></li>
          
          <li><a href="/post/2018-01-07.%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0%E4%B8%83/">生物信息学学习笔记（七）</a></li>
          
        </ul>
      </div>
      
    

    

    
<section id="comments">
  
    
<div id="disqus_thread"></div>
<script>
  let disqus_config = function () {
    
    
    
  };
  (function() {
    if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
      document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
      return;
    }
    var d = document, s = d.createElement('script'); s.async = true;
    s.src = 'https://' + "logos-1" + '.disqus.com/embed.js';
    s.setAttribute('data-timestamp', +new Date());
    (d.head || d.body).appendChild(s);
  })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>


  
</section>



  </div>
</article>

      

    
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>

      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/highlight.min.js" integrity="sha256-aYTdUrn6Ow1DDgh5JTc3aDGnnju48y/1c8s1dgkYPQ8=" crossorigin="anonymous"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/languages/r.min.js"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/languages/css.min.js"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/languages/bash.min.js"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/languages/js.min.js"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/languages/markdown.min.js"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/languages/html.min.js"></script>
        
      

      
      
    

    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.2.0/leaflet.js" integrity="sha512-lInM/apFSqyy1o6s89K4iQUKg6ppXEgsVxT35HbzUupEVRh2Eu9Wdl4tHj7dZO0s1uvplcYGmt3498TtHq+log==" crossorigin="anonymous"></script>
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    
    <script id="dsq-count-scr" src="https://logos-1.disqus.com/count.js" async></script>
    

    
    
    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.16bbb3750feb7244c9bc409a5a4fe678.js"></script>

    






  
  <div class="container">
    <footer class="site-footer">
  

  <p class="powered-by">
    &copy; Tao Yan, 2018-2019 &middot; 

    Made with <i class="far fa-heart" style="color:#E13D3D"></i>, <a href="https://github.com/YTLogos/mywebsite" target="_blank" rel="noopener"><i class="fas fa-code-branch" style="color:#E13D3D"></i></a>, the <a href="https://cran.r-project.org/" target="_blank" rel="noopener"><i class="fab fa-r-project"></i> </a><a href="https://github.com/rstudio/blogdown" target="_blank" rel="noopener">blogdown</a> package, and the 
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" id="back_to_top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

<div style="text-align: center;"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5cz8zk3uynm&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script></div>

  </div>
  

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

</body>
</html>
