<!DOCTYPE html><html lang="en-us" >

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 4.8.0">

  

  
  
  
  
  
    
    
    
  
  

  

  
  
  
    
  
  <meta name="description" content="
简介
系统发育树是一种推断各种生物之间进化关系的好方法，在进化研究中得到了广泛的应用，得益于测序技术的发展以及成本的不断下降，大量的物种以及群体被测序，产生了海量的基因型数据，在重测序项目中，基于SNP数据进行系统发育树的构建有利于更全面地囊括整个基因组层面的变异进行分析。
SNPhylo是基于SNP数据构建群体系统发育树的一个pipeline，可以接受主流的VCF文件以及HapMap文件，同时其内置了过滤功能，通过对低质量的SNP，连锁不平衡进行过滤，生成用于构建群体系统发育树的输入数据，SNPhylo会调用muscle进行对序列比对，DNAML构树，具体流程如下：">

  
  <link rel="alternate" hreflang="en-us" href="https://taoyan.netlify.app/post/2020-07-15.%E4%BD%BF%E7%94%A8snphylo%E5%9F%BA%E4%BA%8Esnp%E6%9E%84%E5%BB%BA%E7%BE%A4%E4%BD%93%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%E6%A0%91/">

  


  
  
  
  <meta name="theme-color" content="#06805E">
  

  
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.0-1/css/all.min.css" integrity="sha256-4w9DunooKSr3MFXHXWyFER38WmPdm361bQS/2KUWZbU=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css" integrity="sha256-Vzbj7sDDS/woiFS3uNKo8eIuni59rjyNGtXfstRzStA=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/tomorrow.min.css" crossorigin="anonymous" title="hl-light">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/tomorrow.min.css" crossorigin="anonymous" title="hl-dark" disabled>
      
    

    

    

    
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js" integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin="anonymous" async></script>
      
    
      

      
      

      
    
      

      
      

      
    
      

      
      
        
      

      
    
      

      
      

      
    

  

  
  
  
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lora:400,400i%7CKarla:400,400i,700,700i%7CRoboto+Mono%7CKaushan+Script&display=swap">
  

  
  
  
  
  <link rel="stylesheet" href="/css/academic.css">

  





<script async src="https://www.googletagmanager.com/gtag/js?id=UA-99209585-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];

  function gtag() {
      dataLayer.push(arguments);
  }

  function trackOutboundLink(url, target) {
    gtag('event', 'click', {
         'event_category': 'outbound',
         'event_label': url,
         'transport_type': 'beacon',
         'event_callback': function () {
           if (target !== '_blank') {
             document.location = url;
           }
         }
    });
    console.debug("Outbound link clicked: " + url);
  }

  function onClickCallback(event) {
    if ((event.target.tagName !== 'A') || (event.target.host === window.location.host)) {
      return;
    }
    trackOutboundLink(event.target, event.target.getAttribute('target'));  
  }

  gtag('js', new Date());
  gtag('config', 'UA-99209585-1', {});

  
  document.addEventListener('click', onClickCallback, false);
</script>


  


  
  

  

  <link rel="manifest" href="/index.webmanifest">
  <link rel="icon" type="image/png" href="/images/icon_hubd7880285348c4660acc9402140a2483_242472_32x32_fill_lanczos_center_2.png">
  <link rel="apple-touch-icon" type="image/png" href="/images/icon_hubd7880285348c4660acc9402140a2483_242472_192x192_fill_lanczos_center_2.png">

  <link rel="canonical" href="https://taoyan.netlify.app/post/2020-07-15.%E4%BD%BF%E7%94%A8snphylo%E5%9F%BA%E4%BA%8Esnp%E6%9E%84%E5%BB%BA%E7%BE%A4%E4%BD%93%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%E6%A0%91/">

  
  
  
  
  
  
  
    
  
  
  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@TaoYANLogos">
  <meta property="twitter:creator" content="@TaoYANLogos">
  
  <meta property="og:site_name" content="Tao Yan">
  <meta property="og:url" content="https://taoyan.netlify.app/post/2020-07-15.%E4%BD%BF%E7%94%A8snphylo%E5%9F%BA%E4%BA%8Esnp%E6%9E%84%E5%BB%BA%E7%BE%A4%E4%BD%93%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%E6%A0%91/">
  <meta property="og:title" content="使用SNPhylo基于SNP构建群体系统发育树 | Tao Yan">
  <meta property="og:description" content="
简介
系统发育树是一种推断各种生物之间进化关系的好方法，在进化研究中得到了广泛的应用，得益于测序技术的发展以及成本的不断下降，大量的物种以及群体被测序，产生了海量的基因型数据，在重测序项目中，基于SNP数据进行系统发育树的构建有利于更全面地囊括整个基因组层面的变异进行分析。
SNPhylo是基于SNP数据构建群体系统发育树的一个pipeline，可以接受主流的VCF文件以及HapMap文件，同时其内置了过滤功能，通过对低质量的SNP，连锁不平衡进行过滤，生成用于构建群体系统发育树的输入数据，SNPhylo会调用muscle进行对序列比对，DNAML构树，具体流程如下："><meta property="og:image" content="https://taoyan.netlify.app/img/pom-card.png">
  <meta property="twitter:image" content="https://taoyan.netlify.app/img/pom-card.png"><meta property="og:locale" content="en-us">
  
    
      <meta property="article:published_time" content="2020-07-15T16:19:05&#43;00:00">
    
    <meta property="article:modified_time" content="2020-07-15T16:19:05&#43;00:00">
  

  


    






  




<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "https://taoyan.netlify.app/post/2020-07-15.%E4%BD%BF%E7%94%A8snphylo%E5%9F%BA%E4%BA%8Esnp%E6%9E%84%E5%BB%BA%E7%BE%A4%E4%BD%93%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%E6%A0%91/"
  },
  "headline": "使用SNPhylo基于SNP构建群体系统发育树",
  
  "datePublished": "2020-07-15T16:19:05Z",
  "dateModified": "2020-07-15T16:19:05Z",
  
  "publisher": {
    "@type": "Organization",
    "name": "Tao Yan",
    "logo": {
      "@type": "ImageObject",
      "url": "https://taoyan.netlify.app/images/icon_hubd7880285348c4660acc9402140a2483_242472_192x192_fill_lanczos_center_2.png"
    }
  },
  "description": "\u003cp\u003e\u003cimg src=\"https://raw.githubusercontent.com/YTLogos/pic_link/master/img/20201006190614.png\" alt=\"20201006190614.png\"\u003e\u003c/p\u003e\n\u003ch2 id=\"简介\"\u003e简介\u003c/h2\u003e\n\u003cp\u003e系统发育树是一种推断各种生物之间进化关系的好方法，在进化研究中得到了广泛的应用，得益于测序技术的发展以及成本的不断下降，大量的物种以及群体被测序，产生了海量的基因型数据，在重测序项目中，基于SNP数据进行系统发育树的构建有利于更全面地囊括整个基因组层面的变异进行分析。\n\u003cstrong\u003eSNPhylo\u003c/strong\u003e是基于SNP数据构建群体系统发育树的一个pipeline，可以接受主流的VCF文件以及HapMap文件，同时其内置了过滤功能，通过对低质量的SNP，连锁不平衡进行过滤，生成用于构建群体系统发育树的输入数据，\u003cstrong\u003eSNPhylo\u003c/strong\u003e会调用\u003cstrong\u003emuscle\u003c/strong\u003e进行对序列比对，\u003cstrong\u003eDNAML\u003c/strong\u003e构树，具体流程如下：\u003c/p\u003e"
}
</script>

  

  


  


  





  <title>使用SNPhylo基于SNP构建群体系统发育树 | Tao Yan</title>

</head>

<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" >

  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


  
<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">Tao Yan</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav ml-auto">
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link  active" href="/post"><span>Blog</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/publication"><span>Publications</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/projects"><span>Projects</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/CV.html"><span>CV</span></a>
        </li>

        
        

      

        

        
        <li class="nav-item">
          <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        

        

      </ul>

    </div>
  </div>
</nav>


  <article class="article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1>使用SNPhylo基于SNP构建群体系统发育树</h1>

  

  
    



<meta content="2020-07-15 16:19:05 &#43;0000 UTC" itemprop="datePublished">
<meta content="2020-07-15 16:19:05 &#43;0000 UTC" itemprop="dateModified">

<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
    
    <time>Jul 15, 2020</time>
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    3 min read
  </span>
  

  
  

  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fas fa-folder"></i>
    <a href="/categories/bioinformatics/">bioinformatics</a></span>
  

  
    
<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=https://taoyan.netlify.app/post/2020-07-15.%E4%BD%BF%E7%94%A8snphylo%E5%9F%BA%E4%BA%8Esnp%E6%9E%84%E5%BB%BA%E7%BE%A4%E4%BD%93%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%E6%A0%91/&amp;text=%e4%bd%bf%e7%94%a8SNPhylo%e5%9f%ba%e4%ba%8eSNP%e6%9e%84%e5%bb%ba%e7%be%a4%e4%bd%93%e7%b3%bb%e7%bb%9f%e5%8f%91%e8%82%b2%e6%a0%91" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=https://taoyan.netlify.app/post/2020-07-15.%E4%BD%BF%E7%94%A8snphylo%E5%9F%BA%E4%BA%8Esnp%E6%9E%84%E5%BB%BA%E7%BE%A4%E4%BD%93%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%E6%A0%91/&amp;t=%e4%bd%bf%e7%94%a8SNPhylo%e5%9f%ba%e4%ba%8eSNP%e6%9e%84%e5%bb%ba%e7%be%a4%e4%bd%93%e7%b3%bb%e7%bb%9f%e5%8f%91%e8%82%b2%e6%a0%91" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=%e4%bd%bf%e7%94%a8SNPhylo%e5%9f%ba%e4%ba%8eSNP%e6%9e%84%e5%bb%ba%e7%be%a4%e4%bd%93%e7%b3%bb%e7%bb%9f%e5%8f%91%e8%82%b2%e6%a0%91&amp;body=https://taoyan.netlify.app/post/2020-07-15.%E4%BD%BF%E7%94%A8snphylo%E5%9F%BA%E4%BA%8Esnp%E6%9E%84%E5%BB%BA%E7%BE%A4%E4%BD%93%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%E6%A0%91/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=https://taoyan.netlify.app/post/2020-07-15.%E4%BD%BF%E7%94%A8snphylo%E5%9F%BA%E4%BA%8Esnp%E6%9E%84%E5%BB%BA%E7%BE%A4%E4%BD%93%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%E6%A0%91/&amp;title=%e4%bd%bf%e7%94%a8SNPhylo%e5%9f%ba%e4%ba%8eSNP%e6%9e%84%e5%bb%ba%e7%be%a4%e4%bd%93%e7%b3%bb%e7%bb%9f%e5%8f%91%e8%82%b2%e6%a0%91" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
  </ul>
</div>


  

</div>
    














  
</div>



  <div class="article-container">

    <div class="article-style">
      <p><img src="https://raw.githubusercontent.com/YTLogos/pic_link/master/img/20201006190614.png" alt="20201006190614.png"></p>
<h2 id="简介">简介</h2>
<p>系统发育树是一种推断各种生物之间进化关系的好方法，在进化研究中得到了广泛的应用，得益于测序技术的发展以及成本的不断下降，大量的物种以及群体被测序，产生了海量的基因型数据，在重测序项目中，基于SNP数据进行系统发育树的构建有利于更全面地囊括整个基因组层面的变异进行分析。
<strong>SNPhylo</strong>是基于SNP数据构建群体系统发育树的一个pipeline，可以接受主流的VCF文件以及HapMap文件，同时其内置了过滤功能，通过对低质量的SNP，连锁不平衡进行过滤，生成用于构建群体系统发育树的输入数据，<strong>SNPhylo</strong>会调用<strong>muscle</strong>进行对序列比对，<strong>DNAML</strong>构树，具体流程如下：</p>
<p><img src="https://raw.githubusercontent.com/YTLogos/pic_link/master/img/20201006190614.png" alt="20201006190614.png"></p>
<p><strong>SNPhylo</strong>有以下特性：</p>
<ul>
<li>
<p>基于全基因组SNP的树状结构。传统的建树方法是基于单拷贝基因，核糖体RNA基因，内部转录间隔序列(ITS)等具有特定属性的基因，而<strong>SNPhylo</strong>则是从全基因组层面构建进化树，因此更加准确；</p>
</li>
<li>
<p>通过连锁不平衡减少SNP的冗余。同一<strong>LD Block</strong>中的SNP提供了冗余的系谱信息，<strong>SNPhylo</strong>在一个<strong>LD Block</strong>只保留一个有效的SNP，在不丢失信息位点的情况下，大大减少了运行时间；</p>
</li>
<li>
<p>树的构建过程是高度自动化的。<strong>SNPhylo</strong>可以接受主流的VCF文件作为直接输入数据，一行命令就可以生成最大似然树。</p>
</li>
</ul>
<p><strong>SNPhylo</strong>的源代码在https://github.com/thlee/SNPhylo</p>
<h2 id="数据准备">数据准备</h2>
<p><strong>SNPhylo</strong>内置了一系列的过滤功能，但是我倾向于自己先过滤数据，这样后续<strong>SNPhylo</strong>运行时间也快些。</p>
<h3 id="maf以及geno过滤">maf以及geno过滤</h3>
<p>过滤之前先处理一下基因型数据，很多软件是为人体研究设计的，不支持带有特殊字符的染色体名称，比如chr1，因此第一步我们先修改染色体名称为数值型，这里我们利用<strong>bcftools</strong>:</p>
<pre><code># 先利用bgzip压缩，tabix构建索引
bgzip &lt; original.vcf &gt; original.vcf.gz
tabix -p vcf original.vcf.gz
# 制作染色体名称对应文件
echo &quot;chr1 1&quot; &gt;&gt; chr_name_conv.txt
echo &quot;chr2 2&quot; &gt;&gt; chr_name_conv.txt
....

## 染色体名称修改
bcftools annotate --rename-chrs chr_name_conv.txt original.vcf.gz | bgzip &gt; rename.vcf.gz
tabix -p vcf rename.vcf.gz

## 添加SNP ID， GATK出来的VCF文件是没有SNP ID的，这里还是利用bcftools添加ID
bcftools annotate --set-id '%CHROM\_%POS\_%REF\_%FIRST_ALT' rename.vcf.gz |bgzip &gt; rename.id.vcf.gz
tabix -p vcf rename.id.vcf.gz
</code></pre>
<p>only include SNPs with MAF &gt;= 0.05 and include only SNPs with a 90% genotyping rate (10% missing) use</p>
<pre><code>plink --vcf rename.id.vcf.gz --maf 0.05 --geno 0.1 --recode  vcf-iid --out rename.id.maf0.05.geno0.1
</code></pre>
<p>下面根据LD进行过滤并提取符合条件的SNP用于建树</p>
<pre><code>## LD标记
plink --vcf rename.id.maf0.05.geno0.1.vcf  --indep-pairwise 50 10 0.2 --out rename.id.maf0.05.geno0.1
## LD过滤提取
plink --extract rename.id.maf0.05.geno0.1.prune.in --make-bed --out rename.id.maf0.05.geno0.1.prune.in --recode vcf-iid --vcf rename.id.maf0.05.geno0.1.vcf
</code></pre>
<p>此时我们得到了rename.id.maf0.05.geno0.1.prune.in.vcf用作<strong>SNPhylo</strong>的输入数据。</p>
<h2 id="构建进化树">构建进化树</h2>
<p><strong>SNPhylo</strong>需要安装一些依赖：</p>
<ul>
<li>R包（getopt, phangorn, gdsfmt,SNPRelate）：用于过滤等</li>
<li>python2.7</li>
<li>muscle：序列联配，可选</li>
<li>PHYLIP：构建进化树</li>
</ul>
<p>安装这些环境之后放入环境变量中，或者一会配置的时候手动输入这些软件的位置</p>
<p>此时从GitHub下载最新的<strong>SNPhylo</strong></p>
<pre><code>git clone https://github.com/thlee/SNPhylo.git
</code></pre>
<p>配置：</p>
<pre><code>cd SNPhylo
bash setup.sh

# 配置信息
Version: 20141127

START TO SET UP FOR SNPHYLO!!!

The detected path of R is /usr/bin/R. Is it correct? [Y/n] y
The detected path of python is /usr/bin/python. Is it correct? [Y/n] y
The detected path of muscle is /home/taoyan/biosoft/mybin/muscle. Is it correct? [Y/n] y
The detected path of dnaml is /home/taoyan/biosoft/phylip-3.697/exe/dnaml. Is it correct? [Y/n] y
At least one R package (gdsfmt, SNPRelate, getopt or phangorn) to run this pipeline is not found. Are the packages already installed? [y/N] y
Please enter the directory for R packages (ex: /home/foo/r_packages): /database/R_Library

SNPHYLO is successfully installed!!!
</code></pre>
<p>看一下使用说明：</p>
<pre><code>$ ./snphylo.sh -h
Determine phylogenetic tree based on SNP data with a VCF, a HapMap, a Simple SNP or a GDS file

Version: 20180901

Usage:
	snphylo.sh -v VCF_file [-p Maximum_PLCS (5)] [-c Minimum_depth_of_coverage (5)]|-H HapMap_file [-p Maximum_PNSS (5)]|-s Simple_SNP_file [-p Maximum_PNSS (5)]|-d GDS_file [-l LD_threshold (0.1)] [-m MAF_threshold (0.1)] [-M Missing_rate (0.1)] [-o Outgroup_sample_name] [-P Prefix_of_output_files (snphylo.output)] [-b [-B The_number_of_bootstrap_samples (100)]] [-a The_number_of_the_last_autosome (22)] [-t The_number_of_cores_used (1)] [-r] [-A] [-h]

Options:
	-A: Perform multiple alignment by MUSCLE
	-b: Perform (non-parametric) bootstrap analysis and generate a tree
	-h: Show help and exit
	-r: Skip the step removing low quality data (-p and -c option are ignored).

Acronyms:
	PLCS: The percent of Low Coverage Sample
	PNSS: The percent of Sample which has no SNP information
	LD: Linkage Disequilibrium
	MAF: Minor Allele Frequency

Simple SNP File Format:
	#Chrom	Pos	SampleID1	SampleID2	SampleID3	...
	1	1000	A	A	T	...
	1	1002	G	C	G	...
	...
	2	2000	G	C	G	...
	2	2002	A	A	T	...
	...
</code></pre>
<p>很简单，直接将准备好的rename.id.maf0.05.geno0.1.prune.in.vcf输进去记好了(我这里无需过滤了，所以设置了参数-r)：</p>
<pre><code>./snphylo.sh -v rename.id.maf0.05.geno0.1.prune.in.vcf -A -b -B 1000 -r
</code></pre>
<p>运行结束后回生成一系列文件：</p>
<pre><code>snphylo.output.bs.png
snphylo.output.bs.tree
snphylo.output.fasta
snphylo.output.filtered.vcf
snphylo.output.gds
snphylo.output.id.txt
snphylo.output.ml.png
snphylo.output.ml.tree
snphylo.output.ml.txt
snphylo.output.phylip.txt
snphylo.tar.gz
</code></pre>
<p>其中有树文件，图片也有，但是丑，以前有iTOL，现在收费了，还是老老实实用Y叔的<strong>ggtree</strong>自己绘制美化吧</p>
    </div>

    






<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/phylogenetic/">phylogenetic</a>
  
  <a class="badge badge-light" href="/tags/snps/">SNPs</a>
  
</div>



<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=https://taoyan.netlify.app/post/2020-07-15.%E4%BD%BF%E7%94%A8snphylo%E5%9F%BA%E4%BA%8Esnp%E6%9E%84%E5%BB%BA%E7%BE%A4%E4%BD%93%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%E6%A0%91/&amp;text=%e4%bd%bf%e7%94%a8SNPhylo%e5%9f%ba%e4%ba%8eSNP%e6%9e%84%e5%bb%ba%e7%be%a4%e4%bd%93%e7%b3%bb%e7%bb%9f%e5%8f%91%e8%82%b2%e6%a0%91" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=https://taoyan.netlify.app/post/2020-07-15.%E4%BD%BF%E7%94%A8snphylo%E5%9F%BA%E4%BA%8Esnp%E6%9E%84%E5%BB%BA%E7%BE%A4%E4%BD%93%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%E6%A0%91/&amp;t=%e4%bd%bf%e7%94%a8SNPhylo%e5%9f%ba%e4%ba%8eSNP%e6%9e%84%e5%bb%ba%e7%be%a4%e4%bd%93%e7%b3%bb%e7%bb%9f%e5%8f%91%e8%82%b2%e6%a0%91" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=%e4%bd%bf%e7%94%a8SNPhylo%e5%9f%ba%e4%ba%8eSNP%e6%9e%84%e5%bb%ba%e7%be%a4%e4%bd%93%e7%b3%bb%e7%bb%9f%e5%8f%91%e8%82%b2%e6%a0%91&amp;body=https://taoyan.netlify.app/post/2020-07-15.%E4%BD%BF%E7%94%A8snphylo%E5%9F%BA%E4%BA%8Esnp%E6%9E%84%E5%BB%BA%E7%BE%A4%E4%BD%93%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%E6%A0%91/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=https://taoyan.netlify.app/post/2020-07-15.%E4%BD%BF%E7%94%A8snphylo%E5%9F%BA%E4%BA%8Esnp%E6%9E%84%E5%BB%BA%E7%BE%A4%E4%BD%93%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%E6%A0%91/&amp;title=%e4%bd%bf%e7%94%a8SNPhylo%e5%9f%ba%e4%ba%8eSNP%e6%9e%84%e5%bb%ba%e7%be%a4%e4%bd%93%e7%b3%bb%e7%bb%9f%e5%8f%91%e8%82%b2%e6%a0%91" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
  </ul>
</div>












  
  





  
    
    
    
      
    
    
    
    <div class="media author-card content-widget-hr">
      
        
        <img class="avatar mr-3 avatar-circle" src="/authors/alison/avatar_hufa245e75a5e5fa47274790393ba89624_26717_270x270_fill_q90_lanczos_center.jpg" alt="">
      

      <div class="media-body">
        <h5 class="card-title"><a href="https://taoyan.netlify.app/"></a></h5>
        <h6 class="card-subtitle">PhD Candidate of Crop Genetics and Breeding</h6>
        <p class="card-text">I am a PhD student of Crop Genetics and Breeding at the <em>Zhejiang University</em> Crop Science Lab. My research interests covers a range of issues:Population Genetics Evolution and Ecotype Divergence Analysis of Oilseed Rape, Genome-wide Association Study (GWAS) of Agronomic Traits.</p>
        <ul class="network-icon" aria-hidden="true">
  
    
    
    
      
    
    
    
    
    
    <li>
      <a href="mailto:tyan@zju.edu.cn" >
        <i class="fas fa-paper-plane"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
    <li>
      <a href="mailto:tyan@zju.edu.cn" >
        <i class="fas fa-envelope"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://twitter.com/TaoYANLogos" target="_blank" rel="noopener">
        <i class="fab fa-twitter"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://github.com/YTLogos" target="_blank" rel="noopener">
        <i class="fab fa-github"></i>
      </a>
    </li>
  
    
    
    
    
    
    
    
      
    
    <li>
      <a href="https://scholar.google.com/citations?hl=en&amp;user=8p3wQNwAAAAJ" target="_blank" rel="noopener">
        <i class="ai ai-google-scholar"></i>
      </a>
    </li>
  
</ul>

      </div>
    </div>
  







<section id="comments">
  
    
<div id="disqus_thread"></div>
<script>
  let disqus_config = function () {
    
    
    
  };
  (function() {
    if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
      document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
      return;
    }
    var d = document, s = d.createElement('script'); s.async = true;
    s.src = 'https://' + "logos-1" + '.disqus.com/embed.js';
    s.setAttribute('data-timestamp', +new Date());
    (d.head || d.body).appendChild(s);
  })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>


  
</section>










  </div>
</article>

      

    
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js" integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin="anonymous"></script>

      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js" integrity="sha256-eOgo0OtLL4cdq7RdwRUiGKLX9XsIJ7nGhWEKbohmVAQ=" crossorigin="anonymous"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/r.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/Shell.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/Python.min.js"></script>
        
      

      
      
    

    
    

    
    
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    
    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.37431be2d92d7fb0160054761ab79602.js"></script>

    






  
  
  <div class="container">
    <footer class="site-footer">
  
  <p class="powered-by">
    <a href="/license/">LICENSE: CC-BY-SA <br> <i class="fab fa-creative-commons fa-2x"></i><i class="fab fa-creative-commons-by fa-2x"></i><i class="fab fa-creative-commons-sa fa-2x"></i> </a><br>
  </p>
  

  <p class="powered-by">
    © Tao Yan, 2018-2020 &middot; 

    Made with <i class="far fa-heart" style="color:#E13D3D"></i>, <a href="https://github.com/YTLogos/TaoYan" target="_blank" rel="noopener"><i class="fas fa-code-branch" style="color:#E13D3D"></i></a>, the <a href="https://cran.r-project.org/" target="_blank" rel="noopener"><i class="fab fa-r-project"></i> </a><a href="https://github.com/rstudio/blogdown" target="_blank" rel="noopener">blogdown</a> package, and the 
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" id="back_to_top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

<div style="text-align: center;"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/6.js?i=5cz8zk3uynm&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=0" async="async"></script></div>

  </div>
  

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

</body>
</html>
